{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 387,
   "id": "6e6aa994",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os \n",
    "import json\n",
    "import requests\n",
    "from Bio import Entrez, SeqIO\n",
    "from Bio.Blast import NCBIWWW, NCBIXML"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa797e64",
   "metadata": {},
   "source": [
    "### All phenotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b270b4e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "      <th>0</th>\n",
       "      <th>all_samples</th>\n",
       "      <th>1</th>\n",
       "      <th>processed_runs</th>\n",
       "      <th>2</th>\n",
       "      <th>valid_runs</th>\n",
       "      <th>3</th>\n",
       "      <th>failed_runs</th>\n",
       "      <th>4</th>\n",
       "      <th>nr_species</th>\n",
       "      <th>5</th>\n",
       "      <th>nr_genus</th>\n",
       "      <th>6</th>\n",
       "      <th>term</th>\n",
       "      <th>7</th>\n",
       "      <th>note</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C537163</td>\n",
       "      <td>C537163</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Pediatric Autoimmune Neuropsychiatric Disorder...</td>\n",
       "      <td>Pediatric Autoimmune Neuropsychiatric Disorder...</td>\n",
       "      <td>OBSESSIVE-COMPULSIVE DISORDER and TIC DISORDER...</td>\n",
       "      <td>OBSESSIVE-COMPULSIVE DISORDER and TIC DISORDER...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>D000066891</td>\n",
       "      <td>D000066891</td>\n",
       "      <td>81</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Critical Care Outcomes</td>\n",
       "      <td>Critical Care Outcomes</td>\n",
       "      <td>A measure of the mortality and morbidity rates...</td>\n",
       "      <td>A measure of the mortality and morbidity rates...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>D000067011</td>\n",
       "      <td>D000067011</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1717</td>\n",
       "      <td>1717</td>\n",
       "      <td>663</td>\n",
       "      <td>663</td>\n",
       "      <td>Severe Acute Malnutrition</td>\n",
       "      <td>Severe Acute Malnutrition</td>\n",
       "      <td>Acute form of MALNUTRITION which usually affec...</td>\n",
       "      <td>Acute form of MALNUTRITION which usually affec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D000067877</td>\n",
       "      <td>D000067877</td>\n",
       "      <td>1217</td>\n",
       "      <td>1217</td>\n",
       "      <td>710</td>\n",
       "      <td>710</td>\n",
       "      <td>606</td>\n",
       "      <td>606</td>\n",
       "      <td>104</td>\n",
       "      <td>104</td>\n",
       "      <td>2325</td>\n",
       "      <td>2325</td>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>Autism Spectrum Disorder</td>\n",
       "      <td>Autism Spectrum Disorder</td>\n",
       "      <td>Wide continuum of associated cognitive and neu...</td>\n",
       "      <td>Wide continuum of associated cognitive and neu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D000069279</td>\n",
       "      <td>D000069279</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Drug Resistant Epilepsy</td>\n",
       "      <td>Drug Resistant Epilepsy</td>\n",
       "      <td>Epileptic condition in which adequate trials o...</td>\n",
       "      <td>Epileptic condition in which adequate trials o...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>D056865</td>\n",
       "      <td>D056865</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Ideal Body Weight</td>\n",
       "      <td>Ideal Body Weight</td>\n",
       "      <td>Expected weight of a healthy normal individual...</td>\n",
       "      <td>Expected weight of a healthy normal individual...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>D058246</td>\n",
       "      <td>D058246</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>211</td>\n",
       "      <td>211</td>\n",
       "      <td>92</td>\n",
       "      <td>92</td>\n",
       "      <td>Prehypertension</td>\n",
       "      <td>Prehypertension</td>\n",
       "      <td>Blood pressure levels that are between normote...</td>\n",
       "      <td>Blood pressure levels that are between normote...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>D060825</td>\n",
       "      <td>D060825</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "      <td>Cognitive Dysfunction</td>\n",
       "      <td>Cognitive Dysfunction</td>\n",
       "      <td>Diminished or impaired mental and/or intellect...</td>\n",
       "      <td>Diminished or impaired mental and/or intellect...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>D065290</td>\n",
       "      <td>D065290</td>\n",
       "      <td>290</td>\n",
       "      <td>290</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Acute-On-Chronic Liver Failure</td>\n",
       "      <td>Acute-On-Chronic Liver Failure</td>\n",
       "      <td>Sudden liver failure in the presence of underl...</td>\n",
       "      <td>Sudden liver failure in the presence of underl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>D065626</td>\n",
       "      <td>D065626</td>\n",
       "      <td>728</td>\n",
       "      <td>728</td>\n",
       "      <td>307</td>\n",
       "      <td>307</td>\n",
       "      <td>226</td>\n",
       "      <td>226</td>\n",
       "      <td>81</td>\n",
       "      <td>81</td>\n",
       "      <td>1065</td>\n",
       "      <td>1065</td>\n",
       "      <td>365</td>\n",
       "      <td>365</td>\n",
       "      <td>Non-alcoholic Fatty Liver Disease</td>\n",
       "      <td>Non-alcoholic Fatty Liver Disease</td>\n",
       "      <td>Fatty liver finding without excessive ALCOHOL ...</td>\n",
       "      <td>Fatty liver finding without excessive ALCOHOL ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        disease           0  all_samples     1  processed_runs    2  \\\n",
       "0       C537163     C537163           30    30              30   30   \n",
       "1    D000066891  D000066891           81    81               0    0   \n",
       "2    D000067011  D000067011           46    46              46   46   \n",
       "3    D000067877  D000067877         1217  1217             710  710   \n",
       "4    D000069279  D000069279           28    28               0    0   \n",
       "..          ...         ...          ...   ...             ...  ...   \n",
       "127     D056865     D056865           24    24               0    0   \n",
       "128     D058246     D058246           56    56              56   56   \n",
       "129     D060825     D060825           62    62              62   62   \n",
       "130     D065290     D065290          290   290               0    0   \n",
       "131     D065626     D065626          728   728             307  307   \n",
       "\n",
       "     valid_runs    3  failed_runs    4  nr_species     5  nr_genus    6  \\\n",
       "0             0    0           30   30           0     0         0    0   \n",
       "1             0    0            0    0           0     0         0    0   \n",
       "2            46   46            0    0        1717  1717       663  663   \n",
       "3           606  606          104  104        2325  2325       796  796   \n",
       "4             0    0            0    0           0     0         0    0   \n",
       "..          ...  ...          ...  ...         ...   ...       ...  ...   \n",
       "127           0    0            0    0           0     0         0    0   \n",
       "128          56   56            0    0         211   211        92   92   \n",
       "129          62   62            0    0           0     0        77   77   \n",
       "130           0    0            0    0           0     0         0    0   \n",
       "131         226  226           81   81        1065  1065       365  365   \n",
       "\n",
       "                                                  term  \\\n",
       "0    Pediatric Autoimmune Neuropsychiatric Disorder...   \n",
       "1                               Critical Care Outcomes   \n",
       "2                            Severe Acute Malnutrition   \n",
       "3                             Autism Spectrum Disorder   \n",
       "4                              Drug Resistant Epilepsy   \n",
       "..                                                 ...   \n",
       "127                                  Ideal Body Weight   \n",
       "128                                    Prehypertension   \n",
       "129                              Cognitive Dysfunction   \n",
       "130                     Acute-On-Chronic Liver Failure   \n",
       "131                  Non-alcoholic Fatty Liver Disease   \n",
       "\n",
       "                                                     7  \\\n",
       "0    Pediatric Autoimmune Neuropsychiatric Disorder...   \n",
       "1                               Critical Care Outcomes   \n",
       "2                            Severe Acute Malnutrition   \n",
       "3                             Autism Spectrum Disorder   \n",
       "4                              Drug Resistant Epilepsy   \n",
       "..                                                 ...   \n",
       "127                                  Ideal Body Weight   \n",
       "128                                    Prehypertension   \n",
       "129                              Cognitive Dysfunction   \n",
       "130                     Acute-On-Chronic Liver Failure   \n",
       "131                  Non-alcoholic Fatty Liver Disease   \n",
       "\n",
       "                                                  note  \\\n",
       "0    OBSESSIVE-COMPULSIVE DISORDER and TIC DISORDER...   \n",
       "1    A measure of the mortality and morbidity rates...   \n",
       "2    Acute form of MALNUTRITION which usually affec...   \n",
       "3    Wide continuum of associated cognitive and neu...   \n",
       "4    Epileptic condition in which adequate trials o...   \n",
       "..                                                 ...   \n",
       "127  Expected weight of a healthy normal individual...   \n",
       "128  Blood pressure levels that are between normote...   \n",
       "129  Diminished or impaired mental and/or intellect...   \n",
       "130  Sudden liver failure in the presence of underl...   \n",
       "131  Fatty liver finding without excessive ALCOHOL ...   \n",
       "\n",
       "                                                     8  \n",
       "0    OBSESSIVE-COMPULSIVE DISORDER and TIC DISORDER...  \n",
       "1    A measure of the mortality and morbidity rates...  \n",
       "2    Acute form of MALNUTRITION which usually affec...  \n",
       "3    Wide continuum of associated cognitive and neu...  \n",
       "4    Epileptic condition in which adequate trials o...  \n",
       "..                                                 ...  \n",
       "127  Expected weight of a healthy normal individual...  \n",
       "128  Blood pressure levels that are between normote...  \n",
       "129  Diminished or impaired mental and/or intellect...  \n",
       "130  Sudden liver failure in the presence of underl...  \n",
       "131  Fatty liver finding without excessive ALCOHOL ...  \n",
       "\n",
       "[132 rows x 18 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### -- get all phenotypes  --\n",
    "url = 'https://gmrepo.humangut.info/api/get_all_phenotypes'\n",
    "pheno_01 = requests.post(url, data={})\n",
    "pheno_01_cont = pheno_01.json().get('phenotypes')\n",
    "\n",
    "## -- a DateFrame --\n",
    "phenotypes = pd.DataFrame(pheno_01_cont)\n",
    "\n",
    "phenotypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8627cc1c",
   "metadata": {},
   "source": [
    "### Statistics on particular phenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4ec280be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>metadata</th>\n",
       "      <th>metadata_obtained</th>\n",
       "      <th>stats</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>uid</th>\n",
       "      <td>D006262</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>term</th>\n",
       "      <td>Health</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>note</th>\n",
       "      <td>The state of the organism when it functions op...</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mappingTo</th>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nr_total_samples</th>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>34019.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nr_loaded_samples</th>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>23907.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nr_valid_samples</th>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>16282.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nr_species</th>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>6206.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nr_genus</th>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>1676.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            metadata  \\\n",
       "uid                                                          D006262   \n",
       "term                                                          Health   \n",
       "note               The state of the organism when it functions op...   \n",
       "mappingTo                                                              \n",
       "nr_total_samples                                                 NaN   \n",
       "nr_loaded_samples                                                NaN   \n",
       "nr_valid_samples                                                 NaN   \n",
       "nr_species                                                       NaN   \n",
       "nr_genus                                                         NaN   \n",
       "\n",
       "                   metadata_obtained    stats  \n",
       "uid                             True      NaN  \n",
       "term                            True      NaN  \n",
       "note                            True      NaN  \n",
       "mappingTo                       True      NaN  \n",
       "nr_total_samples                True  34019.0  \n",
       "nr_loaded_samples               True  23907.0  \n",
       "nr_valid_samples                True  16282.0  \n",
       "nr_species                      True   6206.0  \n",
       "nr_genus                        True   1676.0  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## -- get summary information by mesh_id\n",
    "pheno_02_query = {'mesh_id':'D006262'}  ## -- to get statistics on MeSH ID D006262\n",
    "url = 'https://gmrepo.humangut.info/api/getStatisticsByProjectsByMeshID'\n",
    "pheno_02 = requests.post(url, data=json.dumps(pheno_02_query))\n",
    "pheno_02_cont = pheno_02.json()\n",
    "\n",
    "## --get DataFrame\n",
    "phenotyp_stats = pd.DataFrame(pheno_02.json())\n",
    "\n",
    "phenotyp_stats"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8227a6d1",
   "metadata": {},
   "source": [
    "## **Associated species of a phenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1d9790fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "      <th>taxon_rank_level</th>\n",
       "      <th>ncbi_taxon_id</th>\n",
       "      <th>samples</th>\n",
       "      <th>abus_mean</th>\n",
       "      <th>abus_median</th>\n",
       "      <th>abus_sd</th>\n",
       "      <th>scientific_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>820</td>\n",
       "      <td>12002</td>\n",
       "      <td>4.608746</td>\n",
       "      <td>2.110325</td>\n",
       "      <td>6.386639</td>\n",
       "      <td>Bacteroides uniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>821</td>\n",
       "      <td>11417</td>\n",
       "      <td>4.090026</td>\n",
       "      <td>1.254770</td>\n",
       "      <td>6.860151</td>\n",
       "      <td>Bacteroides vulgatus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>28116</td>\n",
       "      <td>11406</td>\n",
       "      <td>2.305965</td>\n",
       "      <td>0.739456</td>\n",
       "      <td>4.349414</td>\n",
       "      <td>Bacteroides ovatus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>40520</td>\n",
       "      <td>11034</td>\n",
       "      <td>0.780763</td>\n",
       "      <td>0.348089</td>\n",
       "      <td>1.346551</td>\n",
       "      <td>Blautia obeum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>39491</td>\n",
       "      <td>10798</td>\n",
       "      <td>2.758767</td>\n",
       "      <td>0.183908</td>\n",
       "      <td>5.977531</td>\n",
       "      <td>[Eubacterium] rectale</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>67296</td>\n",
       "      <td>2</td>\n",
       "      <td>0.015425</td>\n",
       "      <td>0.015425</td>\n",
       "      <td>0.006480</td>\n",
       "      <td>Streptomyces finlayi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>1263550</td>\n",
       "      <td>2</td>\n",
       "      <td>0.010863</td>\n",
       "      <td>0.010863</td>\n",
       "      <td>0.005858</td>\n",
       "      <td>Edwardsiella piscicida</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>69218</td>\n",
       "      <td>2</td>\n",
       "      <td>0.096905</td>\n",
       "      <td>0.096905</td>\n",
       "      <td>0.117344</td>\n",
       "      <td>Enterobacter cancerogenus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>378211</td>\n",
       "      <td>2</td>\n",
       "      <td>0.012495</td>\n",
       "      <td>0.012495</td>\n",
       "      <td>0.016327</td>\n",
       "      <td>Methyloversatilis universalis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>D006262</td>\n",
       "      <td>species</td>\n",
       "      <td>184748</td>\n",
       "      <td>2</td>\n",
       "      <td>0.033364</td>\n",
       "      <td>0.033364</td>\n",
       "      <td>0.046706</td>\n",
       "      <td>Paracoccus seriniphilus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>460 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     disease taxon_rank_level  ncbi_taxon_id  samples  abus_mean  abus_median  \\\n",
       "0    D006262          species            820    12002   4.608746     2.110325   \n",
       "1    D006262          species            821    11417   4.090026     1.254770   \n",
       "2    D006262          species          28116    11406   2.305965     0.739456   \n",
       "3    D006262          species          40520    11034   0.780763     0.348089   \n",
       "4    D006262          species          39491    10798   2.758767     0.183908   \n",
       "..       ...              ...            ...      ...        ...          ...   \n",
       "455  D006262          species          67296        2   0.015425     0.015425   \n",
       "456  D006262          species        1263550        2   0.010863     0.010863   \n",
       "457  D006262          species          69218        2   0.096905     0.096905   \n",
       "458  D006262          species         378211        2   0.012495     0.012495   \n",
       "459  D006262          species         184748        2   0.033364     0.033364   \n",
       "\n",
       "      abus_sd                scientific_name  \n",
       "0    6.386639          Bacteroides uniformis  \n",
       "1    6.860151           Bacteroides vulgatus  \n",
       "2    4.349414             Bacteroides ovatus  \n",
       "3    1.346551                  Blautia obeum  \n",
       "4    5.977531          [Eubacterium] rectale  \n",
       "..        ...                            ...  \n",
       "455  0.006480           Streptomyces finlayi  \n",
       "456  0.005858         Edwardsiella piscicida  \n",
       "457  0.117344      Enterobacter cancerogenus  \n",
       "458  0.016327  Methyloversatilis universalis  \n",
       "459  0.046706        Paracoccus seriniphilus  \n",
       "\n",
       "[460 rows x 8 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno_03_query = {'mesh_id':'D006262'}  ## -- to get statistics on MeSH ID D006262\n",
    "url = 'https://gmrepo.humangut.info/api/getAssociatedSpeciesByMeshID'\n",
    "pheno_03 = requests.post(url, data=json.dumps(pheno_03_query))\n",
    "pheno_03_cont = pheno_03.json()\n",
    "\n",
    "## --get DataFrame\n",
    "phenotyp_assoc_species = pd.DataFrame(pheno_03.json())\n",
    "\n",
    "## --show data header of the resulting DataFrame\n",
    "phenotyp_assoc_species"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63faea14",
   "metadata": {},
   "source": [
    "### Get relative species/genus abundances in samples/runs associated with a phenotype\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "4c0f0e86",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>x</th>\n",
       "      <th>y</th>\n",
       "      <th>counts</th>\n",
       "      <th>cumsum</th>\n",
       "      <th>cumpct</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>89.780469</td>\n",
       "      <td>1186</td>\n",
       "      <td>1186</td>\n",
       "      <td>89.780469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>7.115821</td>\n",
       "      <td>94</td>\n",
       "      <td>1280</td>\n",
       "      <td>96.896291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>1.816805</td>\n",
       "      <td>24</td>\n",
       "      <td>1304</td>\n",
       "      <td>98.713096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7</td>\n",
       "      <td>0.832702</td>\n",
       "      <td>11</td>\n",
       "      <td>1315</td>\n",
       "      <td>99.545799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>0.454201</td>\n",
       "      <td>6</td>\n",
       "      <td>1321</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   x          y  counts  cumsum      cumpct\n",
       "0  1  89.780469    1186    1186   89.780469\n",
       "1  3   7.115821      94    1280   96.896291\n",
       "2  5   1.816805      24    1304   98.713096\n",
       "3  7   0.832702      11    1315   99.545799\n",
       "4  9   0.454201       6    1321  100.000000"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_query = {'mesh_id':'D003093',\"ncbi_taxon_id\" : \"40520\"}  ## -- to get statistics on MeSH ID D006262\n",
    "url = 'https://gmrepo.humangut.info/api/getMicrobeAbundancesByPhenotypeMeshIDAndNCBITaxonID'\n",
    "data = requests.post(url, data=json.dumps(data_query))\n",
    "\n",
    "## --get DataFrames\n",
    "hist_data_for_phenotype = pd.DataFrame(data.json().get('hist_data_for_phenotype'))\n",
    "hist_data_for_phenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "18b3895b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   nr_assoc_runs\n",
      "0          34019\n"
     ]
    }
   ],
   "source": [
    "## -- count associated runs --\n",
    "pheno_06_query = {'mesh_id':'D006262'}  \n",
    "url = 'https://gmrepo.humangut.info/api/countAssociatedRunsByPhenotypeMeshID'\n",
    "pheno_06 = requests.post(url, data=json.dumps(pheno_06_query))\n",
    "pheno_06_cont = pheno_06.json()\n",
    "\n",
    "## -- the resulting variable is a vector --\n",
    "phenotyp_nr_assoc_runs = pd.DataFrame(pheno_06.json())\n",
    "print(phenotyp_nr_assoc_runs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "id": "99e05577",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['checking', 'project_id', 'our_project_id', 'sample_name',\n",
       "       'original_sample_description', 'curated_sample_description', 'run_id',\n",
       "       'sample_id', 'second_sample_id', 'experiment_type',\n",
       "       'nr_reads_sequenced', 'instrument_model', 'disease', 'phenotype',\n",
       "       'is_disease_stage_available', 'disease_stage', 'more', 'more_info',\n",
       "       'country', 'collection_date', 'sex', 'host_age', 'diet', 'longitude',\n",
       "       'latitude', 'BMI', 'Recent.Antibiotics.Use', 'antibiotics_used',\n",
       "       'Antibiotics.Dose', 'Days.Without.Antibiotics.Use', 'accession_id',\n",
       "       'QCStatus'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 366,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## -- get all associted runs --\n",
    "## use skip = 0, limit = 100 to retrieve the first 100 runs, then\n",
    "##     skip = 100, limit = 100 to retrieve the next 100 runs ....\n",
    "\n",
    "pheno_07_query = {'mesh_id':'D006262',\"skip\":0, \"limit\":100}  \n",
    "url = 'https://gmrepo.humangut.info/api/getAssociatedRunsByPhenotypeMeshIDLimit'\n",
    "pheno_07 = requests.post(url, data=json.dumps(pheno_07_query))\n",
    "pheno_07_cont = pheno_07.json()\n",
    "\n",
    "## -- the resulting variable is a vector --\n",
    "phenotyp_a_page_of_assoc_runs = pd.DataFrame(pheno_07.json())\n",
    "phenotyp_a_page_of_assoc_runs.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ca297ee",
   "metadata": {},
   "source": [
    "### Associated genus of a phenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "2f202a12",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease</th>\n",
       "      <th>taxon_rank_level</th>\n",
       "      <th>ncbi_taxon_id</th>\n",
       "      <th>samples</th>\n",
       "      <th>abus_mean</th>\n",
       "      <th>abus_median</th>\n",
       "      <th>abus_sd</th>\n",
       "      <th>scientific_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>816</td>\n",
       "      <td>15443</td>\n",
       "      <td>24.018719</td>\n",
       "      <td>17.775100</td>\n",
       "      <td>21.844879</td>\n",
       "      <td>Bacteroides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>216851</td>\n",
       "      <td>13731</td>\n",
       "      <td>6.127041</td>\n",
       "      <td>3.857800</td>\n",
       "      <td>7.454345</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>572511</td>\n",
       "      <td>13690</td>\n",
       "      <td>2.587185</td>\n",
       "      <td>1.266020</td>\n",
       "      <td>4.440271</td>\n",
       "      <td>Blautia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>1678</td>\n",
       "      <td>13523</td>\n",
       "      <td>8.015112</td>\n",
       "      <td>1.343310</td>\n",
       "      <td>17.006023</td>\n",
       "      <td>Bifidobacterium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>841</td>\n",
       "      <td>13258</td>\n",
       "      <td>2.248774</td>\n",
       "      <td>0.943384</td>\n",
       "      <td>4.266204</td>\n",
       "      <td>Roseburia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>991903</td>\n",
       "      <td>2</td>\n",
       "      <td>0.057874</td>\n",
       "      <td>0.057874</td>\n",
       "      <td>0.046693</td>\n",
       "      <td>Polymorphum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>44258</td>\n",
       "      <td>2</td>\n",
       "      <td>0.054114</td>\n",
       "      <td>0.054114</td>\n",
       "      <td>0.015598</td>\n",
       "      <td>Caloramator</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>92793</td>\n",
       "      <td>2</td>\n",
       "      <td>0.272387</td>\n",
       "      <td>0.272387</td>\n",
       "      <td>0.372208</td>\n",
       "      <td>Aquabacterium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>475087</td>\n",
       "      <td>2</td>\n",
       "      <td>0.011037</td>\n",
       "      <td>0.011037</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>Methanosphaerula</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>D006262</td>\n",
       "      <td>genus</td>\n",
       "      <td>37452</td>\n",
       "      <td>2</td>\n",
       "      <td>0.028030</td>\n",
       "      <td>0.028030</td>\n",
       "      <td>0.018385</td>\n",
       "      <td>unclassified Bacteroidetes (miscellaneous)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>154 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     disease taxon_rank_level  ncbi_taxon_id  samples  abus_mean  abus_median  \\\n",
       "0    D006262            genus            816    15443  24.018719    17.775100   \n",
       "1    D006262            genus         216851    13731   6.127041     3.857800   \n",
       "2    D006262            genus         572511    13690   2.587185     1.266020   \n",
       "3    D006262            genus           1678    13523   8.015112     1.343310   \n",
       "4    D006262            genus            841    13258   2.248774     0.943384   \n",
       "..       ...              ...            ...      ...        ...          ...   \n",
       "149  D006262            genus         991903        2   0.057874     0.057874   \n",
       "150  D006262            genus          44258        2   0.054114     0.054114   \n",
       "151  D006262            genus          92793        2   0.272387     0.272387   \n",
       "152  D006262            genus         475087        2   0.011037     0.011037   \n",
       "153  D006262            genus          37452        2   0.028030     0.028030   \n",
       "\n",
       "       abus_sd                             scientific_name  \n",
       "0    21.844879                                 Bacteroides  \n",
       "1     7.454345                            Faecalibacterium  \n",
       "2     4.440271                                     Blautia  \n",
       "3    17.006023                             Bifidobacterium  \n",
       "4     4.266204                                   Roseburia  \n",
       "..         ...                                         ...  \n",
       "149   0.046693                                 Polymorphum  \n",
       "150   0.015598                                 Caloramator  \n",
       "151   0.372208                               Aquabacterium  \n",
       "152   0.000033                            Methanosphaerula  \n",
       "153   0.018385  unclassified Bacteroidetes (miscellaneous)  \n",
       "\n",
       "[154 rows x 8 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno_04_query = {'mesh_id':'D006262'}  ## -- to get statistics on MeSH ID D006262\n",
    "url = 'https://gmrepo.humangut.info/api/getAssociatedGeneraByMeshID'\n",
    "pheno_04 = requests.post(url, data=json.dumps(pheno_04_query))\n",
    "pheno_04_cont = pheno_04.json()\n",
    "\n",
    "## --get DataFrame\n",
    "phenotyp_assoc_genera = pd.DataFrame(pheno_04.json())\n",
    "\n",
    "## --show data header of the resulting DataFrame\n",
    "phenotyp_assoc_genera"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcaf4c46",
   "metadata": {},
   "source": [
    "## Data Collection\n",
    "\n",
    "### Fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "8cd4ee90",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR10911129.1</td>\n",
       "      <td>(N, T, G, C, C, A, G, C, C, G, C, C, G, C, G, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR10911129.1</td>\n",
       "      <td>(G, G, A, C, T, A, C, T, A, G, G, G, T, T, T, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR10911129.2</td>\n",
       "      <td>(N, T, G, C, C, A, G, C, C, G, C, C, G, C, G, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR10911129.2</td>\n",
       "      <td>(G, G, A, C, T, A, C, C, A, G, G, G, T, T, T, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR10911129.3</td>\n",
       "      <td>(N, T, G, C, C, A, G, C, A, G, C, C, G, C, G, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              ID                                           Sequence\n",
       "0  SRR10911129.1  (N, T, G, C, C, A, G, C, C, G, C, C, G, C, G, ...\n",
       "1  SRR10911129.1  (G, G, A, C, T, A, C, T, A, G, G, G, T, T, T, ...\n",
       "2  SRR10911129.2  (N, T, G, C, C, A, G, C, C, G, C, C, G, C, G, ...\n",
       "3  SRR10911129.2  (G, G, A, C, T, A, C, C, A, G, G, G, T, T, T, ...\n",
       "4  SRR10911129.3  (N, T, G, C, C, A, G, C, A, G, C, C, G, C, G, ..."
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids = []\n",
    "sequences = []\n",
    "\n",
    "fastq_file = \"SRR10911129.fastq\"\n",
    "\n",
    "with open(fastq_file, \"r\") as handle:\n",
    "    for i, record in enumerate(SeqIO.parse(handle, \"fastq\")):\n",
    "        ids.append(record.id)\n",
    "        sequences.append(record.seq)\n",
    "\n",
    "df = pd.DataFrame({'ID': ids, 'Sequence': sequences})\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c9460da4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bacteria Name: gi|380292720|gb|JQ048306.1| Uncultured bacterium clone TA_91 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|939472076|gb|KT294860.1| Uncultured Klebsiella sp. clone M01598_122_000000000-ADV8A_1_1101_17779_24276 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195542990|gb|EU879536.1| Uncultured bacterium clone DE00600H03 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543047|gb|EU879593.1| Uncultured bacterium clone DE00600C01 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|643012907|gb|KF115476.1| Uncultured Klebsiella sp. clone GVYBRHI01DK8B9 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543029|gb|EU879575.1| Uncultured bacterium clone DE00601H04 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543020|gb|EU879566.1| Uncultured bacterium clone DE00600E12 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|814556964|gb|KP873177.1| Uncultured bacterium clone OTU_30 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|1189362544|gb|MF040050.1| Uncultured bacterium clone Otu143 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543056|gb|EU879602.1| Uncultured bacterium clone DE00600A09 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543323|gb|EU879869.1| Uncultured bacterium clone DE00598D03 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543002|gb|EU879548.1| Uncultured bacterium clone DE00600A05 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195542977|gb|EU879523.1| Uncultured bacterium clone DE00600E04 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543060|gb|EU879606.1| Uncultured bacterium clone DE00601D03 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|939469491|gb|KT292275.1| Uncultured Klebsiella sp. clone M01598_122_000000000-ADV8A_1_1101_13074_5668 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|149166402|dbj|AB326543.1| Uncultured bacterium gene for 16S rRNA, partial sequence\n",
      "Bacteria Name: gi|195543010|gb|EU879556.1| Uncultured bacterium clone DE00600E09 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|195543059|gb|EU879605.1| Uncultured bacterium clone DE00600A08 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2259408387|emb|OW840528.1| uncultured fungus genomic DNA sequence contains ITS1\n",
      "Bacteria Name: gi|643012824|gb|KF115393.1| Uncultured Citrobacter sp. clone GVYBRHI01C94RC 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|643012424|gb|KF114993.1| Uncultured Klebsiella sp. clone GVYBRHI01A6H31 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|149166530|dbj|AB326665.1| Uncultured bacterium gene for 16S rRNA, partial sequence\n",
      "Bacteria Name: gi|195543054|gb|EU879600.1| Uncultured bacterium clone DE00600C08 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|380292680|gb|JQ048266.1| Uncultured bacterium clone TA_51 16S ribosomal RNA gene, partial sequence >gi|380292716|gb|JQ048302.1| Uncultured bacterium clone TA_87 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|643012963|gb|KF115532.1| Uncultured Klebsiella sp. clone GVYBRHI01CDANP 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2648066828|gb|CP141861.1| Klebsiella pneumoniae strain KP34_L5_41.22_ST20_NDM1 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2572954120|gb|CP133342.1| Enterobacter hormaechei strain CRECL421 chromosome, complete genome\n",
      "Bacteria Name: gi|2813903677|gb|PQ432932.1| Enterobacter cloacae subsp. cloacae strain BOTRUEML131 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2561756020|gb|OR426179.1| Enterobacter kobei strain CPO 2.234 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2663918417|gb|CP143005.1| Klebsiella pneumoniae strain CRE-16 chromosome, complete genome\n",
      "Bacteria Name: gi|2592248058|gb|OR672575.1| Klebsiella pneumoniae strain Z-C-41B 16S ribosomal RNA gene, partial sequence >gi|2592248059|gb|OR672576.1| Klebsiella pneumoniae strain Z-C-44B 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2244636626|emb|OW969797.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2841654741|gb|CP143745.1| Enterobacter quasiroggenkampii strain ECC-072 chromosome, complete genome\n",
      "Bacteria Name: gi|2500412503|gb|OQ926069.1| Enterobacter asburiae strain 165_LSS 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2621947122|gb|CP138472.1| Klebsiella pneumoniae subsp. pneumoniae strain KP96_L17_12.23_ST6796_NDM5 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2799645513|gb|CP154186.1| Klebsiella pneumoniae strain KP_NORM_BLD_13534 chromosome, complete genome\n",
      "Bacteria Name: gi|2857167861|gb|PQ620143.1| Klebsiella variicola strain KBL1 16S ribosomal RNA gene, partial sequence >gi|2858723577|gb|PQ623089.1| Klebsiella pneumoniae strain KPL1 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2241862929|emb|OW849243.1| Klebsiella pneumoniae isolate 147 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2623950927|gb|OR852812.1| Enterobacter sp. strain MN691854.1:4-842 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2864024026|gb|PQ671053.1| Klebsiella pneumoniae strain CEMTC_4085 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2243882899|emb|OW967899.1| Klebsiella pneumoniae isolate 152 genome assembly, chromosome: main\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|2276892671|gb|CP101560.1| Klebsiella pneumoniae subsp. pneumoniae strain KP67 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|1720095039|gb|CP041639.1| Klebsiella pneumoniae strain PIMB15ND2KP27 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2507130709|gb|CP125343.1| Klebsiella pneumoniae strain WYKP589 chromosome, complete genome\n",
      "Bacteria Name: gi|2515257717|gb|OR098575.1| Enterobacter hormaechei strain NBAIR18 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2668590045|gb|CP143638.1| Klebsiella pneumoniae strain CP-84 chromosome, complete genome\n",
      "Bacteria Name: gi|2804599336|gb|PQ326981.1| Enterobacter cloacae strain KUMBNGBT-113A 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2197885558|gb|CP055066.1| Klebsiella pneumoniae strain SWHE1 chromosome, complete genome\n",
      "Bacteria Name: gi|2702294132|gb|PP527007.1| Klebsiella pneumoniae strain M6 16S ribosomal RNA gene, partial sequence\n",
      "Bacteria Name: gi|2865742529|gb|PQ691682.1| Klebsiella sp. strain BSDF31-11 16S ribosomal RNA gene, partial sequence\n"
     ]
    }
   ],
   "source": [
    "genomic_sequence = df['Sequence'][0]\n",
    "\n",
    "result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", genomic_sequence)\n",
    "\n",
    "blast_records = NCBIXML.parse(result_handle)\n",
    "\n",
    "for blast_record in blast_records:\n",
    "    for alignment in blast_record.alignments:\n",
    "        for hsp in alignment.hsps:\n",
    "            if hsp.expect < 0.05:\n",
    "                print(\"Bacteria Name:\", alignment.title)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0bd4ece",
   "metadata": {},
   "source": [
    "### API - Parkinsons Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "56c4f40b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ERR365949</th>\n",
       "      <th>scientific_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14.12550</td>\n",
       "      <td>Bacteroides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.18770</td>\n",
       "      <td>Prevotella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9.02015</td>\n",
       "      <td>Ruminococcus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8.71108</td>\n",
       "      <td>Faecalibacterium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.24267</td>\n",
       "      <td>Oscillospira</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ERR365949   scientific_name\n",
       "0   14.12550       Bacteroides\n",
       "1   10.18770        Prevotella\n",
       "2    9.02015      Ruminococcus\n",
       "3    8.71108  Faecalibacterium\n",
       "4    5.24267      Oscillospira"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "runid = \"ERR365949\"\n",
    "\n",
    "query = {\"run_id\":runid}  \n",
    "url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "data = requests.post(url, data=json.dumps(query)).json()\n",
    "\n",
    "## --get run List\n",
    "run = data.get(\"run\")\n",
    "\n",
    "species = pd.DataFrame(data.get(\"species\"))\n",
    "genus = pd.DataFrame(data.get(\"genus\"))\n",
    "\n",
    "#species.drop(['ncbi_taxon_id', 'taxon_rank_level'], axis=1, inplace=True)\n",
    "genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "genusrename = genusdrop.rename(columns={'relative_abundance': runid})\n",
    "genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "genusunique.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "656ae697",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Female</td>\n",
       "      <td>21.22</td>\n",
       "      <td>38</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>D008171</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sex    BMI  Age                   Country  Mesh ID\n",
       "0  Female  21.22   38  United States of America  D008171"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data\n",
    "\n",
    "sex = data['run']['sex']\n",
    "phenotypes = data['phenotypes']\n",
    "bmi = data['run']['BMI']\n",
    "age = data['run']['host_age']\n",
    "country = data['run']['country']\n",
    "mesh_id = phenotypes[0]['disease']\n",
    "\n",
    "health = pd.DataFrame({'Sex': [sex], 'BMI': [bmi], 'Age': [age], 'Country': [country], 'Mesh ID': [mesh_id]})\n",
    "\n",
    "health\n",
    "\n",
    "##use this and the cell above for run id, phenotypes, gender, bmi, weight, country, and taxonomic profile "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "3dfa1888",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "run_id_file = pd.read_csv('Project data files/user_selected_unique_run_ids_TWk0lZG1sI.txt', \n",
    "                          header=None, index_col=None)\n",
    "ERR_run_ids = run_id_file[run_id_file[0].str.startswith('ERR')][0].tolist()\n",
    "ERR_run_idsdf = pd.DataFrame(ERR_run_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "a5f2ba71",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "parkinsons_male_species = []\n",
    "parkinsons_male_health = []\n",
    "\n",
    "for ids in ERR_run_ids: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    mesh_id = phenotypes[0]['disease']\n",
    "    health = pd.DataFrame({'Run ID': ids, 'Sex': [sex], 'BMI': [bmi], 'Age': [age], 'Country': [country], \n",
    "                           'Mesh ID': [mesh_id]})\n",
    "    parkinsons_male_species.append(genusunique)\n",
    "    parkinsons_male_health.append(health)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "c3635c62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR365909</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>61</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR365910</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>53</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR365912</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>66</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR365914</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>63</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR365915</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>72</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Run ID   Sex   BMI  Age  Country  Mesh ID\n",
       "0  ERR365909  Male  None   61  Finland  D010300\n",
       "1  ERR365910  Male  None   53  Finland  D010300\n",
       "2  ERR365912  Male  None   66  Finland  D010300\n",
       "3  ERR365914  Male  None   63  Finland  D010300\n",
       "4  ERR365915  Male  None   72  Finland  D010300"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_health = parkinsons_male_health[0]\n",
    "\n",
    "for df in parkinsons_male_health[1:]:\n",
    "    combined_health = pd.concat([combined_health,df], ignore_index=True)\n",
    "\n",
    "combined_health.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "e47b51d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Adlercreutzia</th>\n",
       "      <th>Aggregatibacter</th>\n",
       "      <th>Akkermansia</th>\n",
       "      <th>Alistipes</th>\n",
       "      <th>Anaerostipes</th>\n",
       "      <th>Anaerotruncus</th>\n",
       "      <th>Bacteroides</th>\n",
       "      <th>Barnesiella</th>\n",
       "      <th>...</th>\n",
       "      <th>Shewanella</th>\n",
       "      <th>Streptococcus</th>\n",
       "      <th>Subdoligranulum</th>\n",
       "      <th>Succinivibrio</th>\n",
       "      <th>Sutterella</th>\n",
       "      <th>Turicibacter</th>\n",
       "      <th>Unknown</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Vibrio</th>\n",
       "      <th>Victivallis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR365909</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.42913</td>\n",
       "      <td>0.138836</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.95153</td>\n",
       "      <td>0.896125</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.1088</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>90.3698</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR365910</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.489687</td>\n",
       "      <td>0.059356</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.3667</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.34783</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.830984</td>\n",
       "      <td>0.14839</td>\n",
       "      <td>109.304</td>\n",
       "      <td>0.192907</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR365912</td>\n",
       "      <td>3.34422</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.34553</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.6336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.052253</td>\n",
       "      <td>10.5291</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.25996</td>\n",
       "      <td>0.07838</td>\n",
       "      <td>21.5545</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR365914</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.191301</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.6629</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.19855</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.48691</td>\n",
       "      <td>0.0</td>\n",
       "      <td>111.72</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR365915</td>\n",
       "      <td>3.72852</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.718898</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43.9381</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.146217</td>\n",
       "      <td>1.63275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.65712</td>\n",
       "      <td>0.0</td>\n",
       "      <td>49.4212</td>\n",
       "      <td>0.085293</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0     Run ID Acidaminococcus Adlercreutzia Aggregatibacter Akkermansia  \\\n",
       "0  ERR365909             0.0           0.0             0.0         0.0   \n",
       "1  ERR365910             0.0           0.0             0.0         0.0   \n",
       "2  ERR365912         3.34422           0.0             0.0         0.0   \n",
       "3  ERR365914             0.0           0.0             0.0         0.0   \n",
       "4  ERR365915         3.72852           0.0             0.0         0.0   \n",
       "\n",
       "0 Alistipes Anaerostipes Anaerotruncus Bacteroides Barnesiella  ...  \\\n",
       "0   0.42913     0.138836           0.0     7.95153    0.896125  ...   \n",
       "1  0.489687     0.059356           0.0     11.3667         0.0  ...   \n",
       "2   1.34553          0.0           0.0     30.6336         0.0  ...   \n",
       "3  0.191301          0.0           0.0     13.6629         0.0  ...   \n",
       "4  0.718898          0.0           0.0     43.9381         0.0  ...   \n",
       "\n",
       "0 Shewanella Streptococcus Subdoligranulum Succinivibrio Sutterella  \\\n",
       "0        0.0           0.0          6.1088           0.0        0.0   \n",
       "1        0.0           0.0         4.34783           0.0   0.830984   \n",
       "2        0.0      0.052253         10.5291           0.0    2.25996   \n",
       "3        0.0           0.0         4.19855           0.0    2.48691   \n",
       "4        0.0      0.146217         1.63275           0.0    1.65712   \n",
       "\n",
       "0 Turicibacter  Unknown Veillonella Vibrio Victivallis  \n",
       "0          0.0  90.3698         0.0    0.0         0.0  \n",
       "1      0.14839  109.304    0.192907    0.0         0.0  \n",
       "2      0.07838  21.5545         0.0    0.0         0.0  \n",
       "3          0.0   111.72         0.0    0.0         0.0  \n",
       "4          0.0  49.4212    0.085293    0.0         0.0  \n",
       "\n",
       "[5 rows x 70 columns]"
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df = parkinsons_male_species[0] \n",
    "\n",
    "for df in parkinsons_male_species[1:]:\n",
    "    combined_df = pd.merge(combined_df, df, on='scientific_name', how='outer')\n",
    "    combined_df.fillna(0, inplace=True)\n",
    "\n",
    "columns_order = ['scientific_name'] + [col for col in combined_df.columns if col != 'scientific_name']\n",
    "combined_df = combined_df[columns_order]\n",
    "\n",
    "transposed_df = combined_df.T \n",
    "\n",
    "\n",
    "dfreset = transposed_df.reset_index()\n",
    "dfreset.columns = dfreset.iloc[0]\n",
    "\n",
    "dffinal = dfreset.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "dffinal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "id": "2eb3ef73",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Adlercreutzia</th>\n",
       "      <th>Aggregatibacter</th>\n",
       "      <th>Akkermansia</th>\n",
       "      <th>...</th>\n",
       "      <th>Shewanella</th>\n",
       "      <th>Streptococcus</th>\n",
       "      <th>Subdoligranulum</th>\n",
       "      <th>Succinivibrio</th>\n",
       "      <th>Sutterella</th>\n",
       "      <th>Turicibacter</th>\n",
       "      <th>Unknown</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Vibrio</th>\n",
       "      <th>Victivallis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR365909</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>61</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.1088</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>90.3698</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR365910</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>53</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.34783</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.830984</td>\n",
       "      <td>0.14839</td>\n",
       "      <td>109.304</td>\n",
       "      <td>0.192907</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR365912</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>66</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>3.34422</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.052253</td>\n",
       "      <td>10.5291</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.25996</td>\n",
       "      <td>0.07838</td>\n",
       "      <td>21.5545</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR365914</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>63</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.19855</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.48691</td>\n",
       "      <td>0.0</td>\n",
       "      <td>111.72</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR365915</td>\n",
       "      <td>Male</td>\n",
       "      <td>None</td>\n",
       "      <td>72</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>3.72852</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.146217</td>\n",
       "      <td>1.63275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.65712</td>\n",
       "      <td>0.0</td>\n",
       "      <td>49.4212</td>\n",
       "      <td>0.085293</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 75 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Run ID   Sex   BMI  Age  Country  Mesh ID Acidaminococcus Adlercreutzia  \\\n",
       "0  ERR365909  Male  None   61  Finland  D010300             0.0           0.0   \n",
       "1  ERR365910  Male  None   53  Finland  D010300             0.0           0.0   \n",
       "2  ERR365912  Male  None   66  Finland  D010300         3.34422           0.0   \n",
       "3  ERR365914  Male  None   63  Finland  D010300             0.0           0.0   \n",
       "4  ERR365915  Male  None   72  Finland  D010300         3.72852           0.0   \n",
       "\n",
       "  Aggregatibacter Akkermansia  ... Shewanella Streptococcus Subdoligranulum  \\\n",
       "0             0.0         0.0  ...        0.0           0.0          6.1088   \n",
       "1             0.0         0.0  ...        0.0           0.0         4.34783   \n",
       "2             0.0         0.0  ...        0.0      0.052253         10.5291   \n",
       "3             0.0         0.0  ...        0.0           0.0         4.19855   \n",
       "4             0.0         0.0  ...        0.0      0.146217         1.63275   \n",
       "\n",
       "  Succinivibrio Sutterella Turicibacter  Unknown Veillonella Vibrio  \\\n",
       "0           0.0        0.0          0.0  90.3698         0.0    0.0   \n",
       "1           0.0   0.830984      0.14839  109.304    0.192907    0.0   \n",
       "2           0.0    2.25996      0.07838  21.5545         0.0    0.0   \n",
       "3           0.0    2.48691          0.0   111.72         0.0    0.0   \n",
       "4           0.0    1.65712          0.0  49.4212    0.085293    0.0   \n",
       "\n",
       "  Victivallis  \n",
       "0         0.0  \n",
       "1         0.0  \n",
       "2         0.0  \n",
       "3         0.0  \n",
       "4         0.0  \n",
       "\n",
       "[5 rows x 75 columns]"
      ]
     },
     "execution_count": 372,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_ids = dffinal['Run ID'].unique()\n",
    "\n",
    "dfmerged = []\n",
    "\n",
    "for run_id in run_ids:\n",
    "    merge = pd.merge(combined_health[combined_health['Run ID'] == run_id], dffinal[dffinal['Run ID'] == run_id], \n",
    "                    on='Run ID', how='inner')\n",
    "    dfmerged.append(merge)\n",
    "    \n",
    "final_p_male_merge = pd.concat(dfmerged, ignore_index=True)\n",
    "\n",
    "final_p_male_merge.to_csv('parkinsons_male.csv', index=False)\n",
    "\n",
    "final_p_male_merge.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5bbc76f4",
   "metadata": {},
   "source": [
    "### API - Parkinsons Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "id": "b0b1c715",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Acidovorax</th>\n",
       "      <th>Actinomyces</th>\n",
       "      <th>Adlercreutzia</th>\n",
       "      <th>...</th>\n",
       "      <th>Streptococcus</th>\n",
       "      <th>Subdoligranulum</th>\n",
       "      <th>Succinatimonas</th>\n",
       "      <th>Succiniclasticum</th>\n",
       "      <th>Sutterella</th>\n",
       "      <th>Turicibacter</th>\n",
       "      <th>Unknown</th>\n",
       "      <th>Varibaculum</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Victivallis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR365911</td>\n",
       "      <td>Female</td>\n",
       "      <td>None</td>\n",
       "      <td>73</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.065172</td>\n",
       "      <td>1.61627</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.86757</td>\n",
       "      <td>0.0</td>\n",
       "      <td>65.0156</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.755996</td>\n",
       "      <td>0.208551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR365913</td>\n",
       "      <td>Female</td>\n",
       "      <td>None</td>\n",
       "      <td>64</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.167344</td>\n",
       "      <td>4.09993</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>95.6969</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR365918</td>\n",
       "      <td>Female</td>\n",
       "      <td>None</td>\n",
       "      <td>73</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.091408</td>\n",
       "      <td>3.4082</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.37112</td>\n",
       "      <td>0.287281</td>\n",
       "      <td>72.6822</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR365919</td>\n",
       "      <td>Female</td>\n",
       "      <td>None</td>\n",
       "      <td>64</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.031803</td>\n",
       "      <td>...</td>\n",
       "      <td>0.87989</td>\n",
       "      <td>8.89431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>55.8465</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR365921</td>\n",
       "      <td>Female</td>\n",
       "      <td>None</td>\n",
       "      <td>66</td>\n",
       "      <td>Finland</td>\n",
       "      <td>D010300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.87146</td>\n",
       "      <td>5.73711</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.43573</td>\n",
       "      <td>0.0</td>\n",
       "      <td>39.5062</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.363108</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Run ID     Sex   BMI  Age  Country  Mesh ID Acidaminococcus Acidovorax  \\\n",
       "0  ERR365911  Female  None   73  Finland  D010300             0.0        0.0   \n",
       "1  ERR365913  Female  None   64  Finland  D010300             0.0        0.0   \n",
       "2  ERR365918  Female  None   73  Finland  D010300             0.0        0.0   \n",
       "3  ERR365919  Female  None   64  Finland  D010300             0.0        0.0   \n",
       "4  ERR365921  Female  None   66  Finland  D010300             0.0        0.0   \n",
       "\n",
       "  Actinomyces Adlercreutzia  ... Streptococcus Subdoligranulum Succinatimonas  \\\n",
       "0         0.0           0.0  ...      0.065172         1.61627            0.0   \n",
       "1         0.0           0.0  ...      0.167344         4.09993            0.0   \n",
       "2         0.0           0.0  ...      0.091408          3.4082            0.0   \n",
       "3         0.0      0.031803  ...       0.87989         8.89431            0.0   \n",
       "4         0.0           0.0  ...       0.87146         5.73711            0.0   \n",
       "\n",
       "  Succiniclasticum Sutterella Turicibacter  Unknown Varibaculum Veillonella  \\\n",
       "0              0.0    2.86757          0.0  65.0156         0.0    0.755996   \n",
       "1              0.0        0.0          0.0  95.6969         0.0         0.0   \n",
       "2              0.0    1.37112     0.287281  72.6822         0.0         0.0   \n",
       "3              0.0        0.0          0.0  55.8465         0.0         0.0   \n",
       "4              0.0    0.43573          0.0  39.5062         0.0    0.363108   \n",
       "\n",
       "  Victivallis  \n",
       "0    0.208551  \n",
       "1         0.0  \n",
       "2         0.0  \n",
       "3         0.0  \n",
       "4         0.0  \n",
       "\n",
       "[5 rows x 72 columns]"
      ]
     },
     "execution_count": 373,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file2 = pd.read_csv('Project data files/user_selected_unique_run_ids_eOkL3H8EJi.txt', \n",
    "                          header=None, index_col=None)\n",
    "ERR_p_female_run_ids = run_id_file2[run_id_file2[0].str.startswith('ERR')][0].tolist()\n",
    "ERR_p_female_run_idsdf = pd.DataFrame(ERR_p_female_run_ids)\n",
    "\n",
    "parkinsons_female_species = []\n",
    "parkinsons_female_health = []\n",
    "\n",
    "for ids in ERR_p_female_run_ids: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    mesh_id = phenotypes[0]['disease']\n",
    "    health = pd.DataFrame({'Run ID': ids, 'Sex': [sex], 'BMI': [bmi], 'Age': [age], 'Country': [country], \n",
    "                           'Mesh ID': [mesh_id]})\n",
    "    parkinsons_female_species.append(genusunique)\n",
    "    parkinsons_female_health.append(health)\n",
    "\n",
    "combined_pfemale_health = parkinsons_female_health[0]\n",
    "\n",
    "for df in parkinsons_female_health[1:]:\n",
    "    combined_pfemale_health = pd.concat([combined_pfemale_health,df], ignore_index=True)\n",
    "\n",
    "\n",
    "combined_pfemale_species = parkinsons_female_species[0] \n",
    "\n",
    "for df in parkinsons_female_species[1:]:\n",
    "    combined_pfemale_species = pd.merge(combined_pfemale_species, df, on='scientific_name', how='outer')\n",
    "    combined_pfemale_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_pfemale = ['scientific_name'] + [col for col in combined_pfemale_species.columns if col != 'scientific_name']\n",
    "combined_pfemale_species = combined_pfemale_species[columns_order_pfemale]\n",
    "\n",
    "transposed_pfemale_df = combined_pfemale_species.T \n",
    "\n",
    "\n",
    "dfreset_pfemale = transposed_pfemale_df.reset_index()\n",
    "dfreset_pfemale.columns = dfreset_pfemale.iloc[0]\n",
    "\n",
    "dffinal_pfemale = dfreset_pfemale.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_pfemale.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids = dffinal_pfemale['Run ID'].unique()\n",
    "\n",
    "dfmerged_pfemale = []\n",
    "\n",
    "for run_id in run_ids:\n",
    "    merge = pd.merge(combined_pfemale_health[combined_pfemale_health['Run ID'] == run_id], \n",
    "                     dffinal_pfemale[dffinal_pfemale['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_pfemale.append(merge)\n",
    "    \n",
    "final_p_female_merge = pd.concat(dfmerged_pfemale, ignore_index=True)\n",
    "\n",
    "final_p_female_merge.to_csv('parkinsons_female.csv', index=False)\n",
    "\n",
    "final_p_female_merge.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d4ecf23",
   "metadata": {},
   "source": [
    "### API - Alzheimers Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "id": "6744f37b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gw/6zvc4vx95yx8vrymkvc5m8zm0000gn/T/ipykernel_49040/2882487975.py:31: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  combined_malz_health = pd.concat([combined_malz_health, df], ignore_index=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acholeplasma</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>...</th>\n",
       "      <th>Thermoanaerobacter</th>\n",
       "      <th>Truepera</th>\n",
       "      <th>Tyzzerella</th>\n",
       "      <th>Uliginosibacterium</th>\n",
       "      <th>Unknown</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Victivallis</th>\n",
       "      <th>Virgibacillus</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Yersinia</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1090510</td>\n",
       "      <td>Male</td>\n",
       "      <td>30.11</td>\n",
       "      <td>31</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D000544]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.008079</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.937222</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.339339</td>\n",
       "      <td>0.032318</td>\n",
       "      <td>18.9626</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1843463</td>\n",
       "      <td>Male</td>\n",
       "      <td>24.41</td>\n",
       "      <td>0</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D000544, D001327]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.036957</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.9544</td>\n",
       "      <td>0.026398</td>\n",
       "      <td>0.042236</td>\n",
       "      <td>0.015839</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1844528</td>\n",
       "      <td>Male</td>\n",
       "      <td>24.41</td>\n",
       "      <td>0</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D000544, D001327]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.027337</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005467</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.1542</td>\n",
       "      <td>0.049207</td>\n",
       "      <td>0.027337</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR2091834</td>\n",
       "      <td>Male</td>\n",
       "      <td>23.72</td>\n",
       "      <td>56</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>[D000544]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.078935</td>\n",
       "      <td>0.036836</td>\n",
       "      <td>...</td>\n",
       "      <td>0.010525</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005262</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.756</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR9671458</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64</td>\n",
       "      <td>China</td>\n",
       "      <td>[D000544]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>183.001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 170 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID   Sex    BMI  Age                   Country             Mesh ID  \\\n",
       "0  ERR1090510  Male  30.11   31  United States of America           [D000544]   \n",
       "1  ERR1843463  Male  24.41    0  United States of America  [D000544, D001327]   \n",
       "2  ERR1844528  Male  24.41    0  United States of America  [D000544, D001327]   \n",
       "3  ERR2091834  Male  23.72   56            United Kingdom           [D000544]   \n",
       "4  SRR9671458  Male    NaN   64                     China           [D000544]   \n",
       "\n",
       "  Abiotrophia Acanthopleuribacter Acholeplasma Acidaminococcus  ...  \\\n",
       "0         0.0            0.008079          0.0        0.937222  ...   \n",
       "1         0.0                 0.0          0.0        0.036957  ...   \n",
       "2         0.0                 0.0          0.0        0.027337  ...   \n",
       "3         0.0                 0.0     0.078935        0.036836  ...   \n",
       "4         0.0                 0.0          0.0             0.0  ...   \n",
       "\n",
       "  Thermoanaerobacter  Truepera Tyzzerella Uliginosibacterium  Unknown  \\\n",
       "0                0.0       0.0   0.339339           0.032318  18.9626   \n",
       "1                0.0       0.0        0.0                0.0  42.9544   \n",
       "2                0.0  0.005467        0.0                0.0  42.1542   \n",
       "3           0.010525       0.0   0.005262                0.0   11.756   \n",
       "4                0.0       0.0        0.0                0.0  183.001   \n",
       "\n",
       "  Veillonella Victivallis Virgibacillus Weissella  Yersinia  \n",
       "0         0.0         0.0           0.0       0.0       0.0  \n",
       "1    0.026398    0.042236      0.015839       0.0       0.0  \n",
       "2    0.049207    0.027337           0.0       0.0  0.005467  \n",
       "3         0.0         0.0           0.0       0.0       0.0  \n",
       "4         0.0         0.0           0.0       0.0       0.0  \n",
       "\n",
       "[5 rows x 170 columns]"
      ]
     },
     "execution_count": 374,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file3 = pd.read_csv('Project data files/user_selected_run_list_eRepc1nyt6.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_alz_male_run_ids = run_id_file3[0]\n",
    "ERR_alz_male_run_ids_drop = ERR_alz_male_run_ids.drop(index=[0, 2, 3, 4])\n",
    "\n",
    "alz_male_species = []\n",
    "alz_male_health = []\n",
    "\n",
    "for ids in ERR_alz_male_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, 'Mesh ID': diseases}])\n",
    "    alz_male_species.append(genusunique)\n",
    "    alz_male_health.append(health)\n",
    "\n",
    "combined_malz_health = alz_male_health[0]\n",
    "\n",
    "for df in alz_male_health[1:]:\n",
    "    combined_malz_health = pd.concat([combined_malz_health, df], ignore_index=True)\n",
    "\n",
    "\n",
    "combined_malz_species = alz_male_species[0] \n",
    "\n",
    "for df1 in alz_male_species[1:]:\n",
    "    combined_malz_species = pd.merge(combined_malz_species, df1, on='scientific_name', how='outer')\n",
    "    combined_malz_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_malz = ['scientific_name'] + [col for col in combined_malz_species.columns if col != 'scientific_name']\n",
    "combined_malz_species = combined_malz_species[columns_order_malz]\n",
    "\n",
    "transposed_malz_df = combined_malz_species.T \n",
    "\n",
    "\n",
    "dfreset_malz = transposed_malz_df.reset_index()\n",
    "dfreset_malz.columns = dfreset_malz.iloc[0]\n",
    "\n",
    "dffinal_malz = dfreset_malz.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_malz.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_malz = dffinal_malz['Run ID'].unique()\n",
    "\n",
    "dfmerged_malz = []\n",
    "\n",
    "for run_id in run_ids_malz:\n",
    "    merge = pd.merge(combined_malz_health[combined_malz_health['Run ID'] == run_id], \n",
    "                     dffinal_malz[dffinal_malz['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_malz.append(merge)\n",
    "    \n",
    "final_malz_merge = pd.concat(dfmerged_malz, ignore_index=True)\n",
    "\n",
    "final_malz_merge.to_csv('alzheimers_male.csv', index=False)\n",
    "\n",
    "final_malz_merge.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c011a58",
   "metadata": {},
   "source": [
    "### API - Alzheimers Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "id": "cebe740c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gw/6zvc4vx95yx8vrymkvc5m8zm0000gn/T/ipykernel_49040/2338621161.py:31: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  combined_femalz_health = pd.concat([combined_femalz_health, df], ignore_index=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Acetobacterium</th>\n",
       "      <th>Acholeplasma</th>\n",
       "      <th>Acidaminococcus</th>\n",
       "      <th>Acidihalobacter</th>\n",
       "      <th>...</th>\n",
       "      <th>Treponema</th>\n",
       "      <th>Tyzzerella</th>\n",
       "      <th>Unknown</th>\n",
       "      <th>Varibaculum</th>\n",
       "      <th>Veillonella</th>\n",
       "      <th>Verrucomicrobium</th>\n",
       "      <th>Vibrio</th>\n",
       "      <th>Victivallis</th>\n",
       "      <th>Viridibacillus</th>\n",
       "      <th>Weissella</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1160800</td>\n",
       "      <td>Female</td>\n",
       "      <td>26.09</td>\n",
       "      <td>53</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D000544, D008881]</td>\n",
       "      <td>0.001165</td>\n",
       "      <td>0.00932</td>\n",
       "      <td>0.33086</td>\n",
       "      <td>0.001165</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001165</td>\n",
       "      <td>40.5105</td>\n",
       "      <td>0.00932</td>\n",
       "      <td>1.00889</td>\n",
       "      <td>0.005825</td>\n",
       "      <td>0.08854</td>\n",
       "      <td>0.00233</td>\n",
       "      <td>0.001165</td>\n",
       "      <td>0.00233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR2091942</td>\n",
       "      <td>Female</td>\n",
       "      <td>25.07</td>\n",
       "      <td>53</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>[D000544, D007410]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.946936</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003573</td>\n",
       "      <td>35.6012</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR9671487</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>67</td>\n",
       "      <td>China</td>\n",
       "      <td>[D000544]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>170.064</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR9671488</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80</td>\n",
       "      <td>China</td>\n",
       "      <td>[D000544]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>114.27</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR9671489</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>57</td>\n",
       "      <td>China</td>\n",
       "      <td>[D000544]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>91.363</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 278 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID     Sex    BMI  Age                   Country  \\\n",
       "0  ERR1160800  Female  26.09   53  United States of America   \n",
       "1  ERR2091942  Female  25.07   53            United Kingdom   \n",
       "2  SRR9671487  Female    NaN   67                     China   \n",
       "3  SRR9671488  Female    NaN   80                     China   \n",
       "4  SRR9671489  Female    NaN   57                     China   \n",
       "\n",
       "              Mesh ID Acetobacterium Acholeplasma Acidaminococcus  \\\n",
       "0  [D000544, D008881]       0.001165      0.00932         0.33086   \n",
       "1  [D000544, D007410]            0.0     0.946936             0.0   \n",
       "2           [D000544]            0.0          0.0             0.0   \n",
       "3           [D000544]            0.0          0.0             0.0   \n",
       "4           [D000544]            0.0          0.0             0.0   \n",
       "\n",
       "  Acidihalobacter  ... Treponema Tyzzerella  Unknown Varibaculum Veillonella  \\\n",
       "0        0.001165  ...       0.0   0.001165  40.5105     0.00932     1.00889   \n",
       "1             0.0  ...       0.0   0.003573  35.6012         0.0         0.0   \n",
       "2             0.0  ...       0.0        0.0  170.064         0.0         0.0   \n",
       "3             0.0  ...       0.0        0.0   114.27         0.0         0.0   \n",
       "4             0.0  ...       0.0        0.0   91.363         0.0         0.0   \n",
       "\n",
       "  Verrucomicrobium   Vibrio Victivallis Viridibacillus Weissella  \n",
       "0         0.005825  0.08854     0.00233       0.001165   0.00233  \n",
       "1              0.0      0.0         0.0            0.0       0.0  \n",
       "2              0.0      0.0         0.0            0.0       0.0  \n",
       "3              0.0      0.0         0.0            0.0       0.0  \n",
       "4              0.0      0.0         0.0            0.0       0.0  \n",
       "\n",
       "[5 rows x 278 columns]"
      ]
     },
     "execution_count": 375,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file4 = pd.read_csv('Project data files/user_selected_run_list_y0G7FDBUha.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_alz_female_run_ids = run_id_file4[0]\n",
    "ERR_alz_female_run_ids_drop = ERR_alz_female_run_ids.drop(index=[0, 1, 3, 4, 5])\n",
    "\n",
    "alz_female_species = []\n",
    "alz_female_health = []\n",
    "\n",
    "for ids in ERR_alz_female_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, 'Mesh ID': diseases}])\n",
    "    alz_female_species.append(genusunique)\n",
    "    alz_female_health.append(health)\n",
    "\n",
    "combined_femalz_health = alz_female_health[0]\n",
    "\n",
    "for df in alz_female_health[1:]:\n",
    "    combined_femalz_health = pd.concat([combined_femalz_health, df], ignore_index=True)\n",
    "\n",
    "\n",
    "combined_femalz_species = alz_female_species[0] \n",
    "\n",
    "for df1 in alz_female_species[1:]:\n",
    "    combined_femalz_species = pd.merge(combined_femalz_species, df1, on='scientific_name', how='outer')\n",
    "    combined_femalz_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_femalz = ['scientific_name'] + [col for col in combined_femalz_species.columns if col != 'scientific_name']\n",
    "combined_femalz_species = combined_femalz_species[columns_order_femalz]\n",
    "\n",
    "transposed_femalz_df = combined_femalz_species.T \n",
    "\n",
    "\n",
    "dfreset_femalz = transposed_femalz_df.reset_index()\n",
    "dfreset_femalz.columns = dfreset_femalz.iloc[0]\n",
    "\n",
    "dffinal_femalz = dfreset_femalz.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_femalz.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_femalz = dffinal_femalz['Run ID'].unique()\n",
    "\n",
    "dfmerged_femalz = []\n",
    "\n",
    "for run_id in run_ids_femalz:\n",
    "    merge = pd.merge(combined_femalz_health[combined_femalz_health['Run ID'] == run_id], \n",
    "                     dffinal_femalz[dffinal_femalz['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_femalz.append(merge)\n",
    "    \n",
    "final_femalz_merge = pd.concat(dfmerged_femalz, ignore_index=True)\n",
    "\n",
    "final_femalz_merge.to_csv('alzheimers_female.csv', index=False)\n",
    "\n",
    "final_femalz_merge.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d2cb15ee",
   "metadata": {},
   "source": [
    "### API - Schizophrenia Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "id": "2db2c2db",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>Acetobacter</th>\n",
       "      <th>...</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Wenyingzhuangia</th>\n",
       "      <th>Wenzhouxiangella</th>\n",
       "      <th>Wolbachia</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Youngiibacter</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1072629</td>\n",
       "      <td>Male</td>\n",
       "      <td>17.67</td>\n",
       "      <td>64</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D007410, D008171, D012559, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00484</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1072937</td>\n",
       "      <td>Male</td>\n",
       "      <td>26.51</td>\n",
       "      <td>53</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D002318, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1073491</td>\n",
       "      <td>Male</td>\n",
       "      <td>28.08</td>\n",
       "      <td>49</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1075554</td>\n",
       "      <td>Male</td>\n",
       "      <td>30.74</td>\n",
       "      <td>62</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001714, D003863, D008881, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1075686</td>\n",
       "      <td>Male</td>\n",
       "      <td>45.84</td>\n",
       "      <td>56</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D008171, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 773 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID   Sex    BMI  Age                   Country  \\\n",
       "0  ERR1072629  Male  17.67   64  United States of America   \n",
       "1  ERR1072937  Male  26.51   53  United States of America   \n",
       "2  ERR1073491  Male  28.08   49  United States of America   \n",
       "3  ERR1075554  Male  30.74   62  United States of America   \n",
       "4  ERR1075686  Male  45.84   56  United States of America   \n",
       "\n",
       "                                             Mesh ID Abiotrophia  \\\n",
       "0  [D001714, D003863, D007410, D008171, D012559, ...         0.0   \n",
       "1               [D001714, D002318, D003863, D012559]         0.0   \n",
       "2                        [D001714, D003863, D012559]    0.003157   \n",
       "3      [D001289, D001714, D003863, D008881, D012559]         0.0   \n",
       "4               [D001714, D003863, D008171, D012559]         0.0   \n",
       "\n",
       "  Acanthopleuribacter Acetoanaerobium Acetobacter  ... Weissella  \\\n",
       "0                 0.0             0.0         0.0  ...       0.0   \n",
       "1                 0.0             0.0         0.0  ...       0.0   \n",
       "2                 0.0             0.0         0.0  ...       0.0   \n",
       "3                 0.0             0.0         0.0  ...       0.0   \n",
       "4                 0.0             0.0         0.0  ...       0.0   \n",
       "\n",
       "  Wenyingzhuangia Wenzhouxiangella Wolbachia Xanthomonas Xenorhabdus Yersinia  \\\n",
       "0             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "1             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "2             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "3             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "4             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "\n",
       "  Youngiibacter Zoogloea Zymomonas  \n",
       "0           0.0  0.00484       0.0  \n",
       "1           0.0      0.0       0.0  \n",
       "2           0.0      0.0       0.0  \n",
       "3           0.0      0.0       0.0  \n",
       "4           0.0      0.0       0.0  \n",
       "\n",
       "[5 rows x 773 columns]"
      ]
     },
     "execution_count": 376,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file5 = pd.read_csv('Project data files/user_selected_run_list_kcYhadsFw3.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_s_male_run_ids = run_id_file5[0]\n",
    "ERR_s_male_run_ids_drop = ERR_s_male_run_ids.drop(index=0)\n",
    "\n",
    "sch_male_species = []\n",
    "sch_male_health = []\n",
    "\n",
    "for ids in ERR_s_male_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    sch_male_species.append(genusunique)\n",
    "    sch_male_health.append(health)\n",
    "\n",
    "combined_msch_health = sch_male_health[0]\n",
    "\n",
    "for df in sch_male_health[1:]:\n",
    "    combined_msch_health = pd.concat([combined_msch_health, df], ignore_index=True)\n",
    "\n",
    "\n",
    "combined_msch_species = sch_male_species[0] \n",
    "\n",
    "for df1 in sch_male_species[1:]:\n",
    "    combined_msch_species = pd.merge(combined_msch_species, df1, on='scientific_name', how='outer')\n",
    "    combined_msch_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_msch = ['scientific_name'] + [col for col in combined_msch_species.columns if col != 'scientific_name']\n",
    "combined_msch_species = combined_msch_species[columns_order_msch]\n",
    "\n",
    "transposed_msch_df = combined_msch_species.T \n",
    "\n",
    "\n",
    "dfreset_msch = transposed_msch_df.reset_index()\n",
    "dfreset_msch.columns = dfreset_msch.iloc[0]\n",
    "\n",
    "dffinal_msch = dfreset_msch.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_msch.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_msch = dffinal_msch['Run ID'].unique()\n",
    "\n",
    "dfmerged_msch = []\n",
    "\n",
    "for run_id in run_ids_msch:\n",
    "    merge = pd.merge(combined_msch_health[combined_msch_health['Run ID'] == run_id], \n",
    "                     dffinal_msch[dffinal_msch['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_msch.append(merge)\n",
    "    \n",
    "final_msch_merge = pd.concat(dfmerged_msch, ignore_index=True)\n",
    "\n",
    "final_msch_merge.to_csv('schizophrenia_male.csv', index=False)\n",
    "\n",
    "final_msch_merge.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38a65f84",
   "metadata": {},
   "source": [
    "### API - Schizophrenia Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 377,
   "id": "c072b1fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetivibrio</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>...</th>\n",
       "      <th>Williamsia</th>\n",
       "      <th>Wolbachia</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenococcus</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Yokenella</th>\n",
       "      <th>Youngiibacter</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1073023</td>\n",
       "      <td>Female</td>\n",
       "      <td>34.72</td>\n",
       "      <td>35</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005583</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1073394</td>\n",
       "      <td>Female</td>\n",
       "      <td>22.41</td>\n",
       "      <td>27</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D001714, D003863, D003920, D003967, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1073395</td>\n",
       "      <td>Female</td>\n",
       "      <td>22.41</td>\n",
       "      <td>27</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D001714, D003863, D003920, D003967, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00247</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1073490</td>\n",
       "      <td>Female</td>\n",
       "      <td>25.73</td>\n",
       "      <td>37</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1073812</td>\n",
       "      <td>Female</td>\n",
       "      <td>29.71</td>\n",
       "      <td>55</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001714, D003015, D003863, D003920, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 930 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID     Sex    BMI  Age                   Country  \\\n",
       "0  ERR1073023  Female  34.72   35  United States of America   \n",
       "1  ERR1073394  Female  22.41   27  United States of America   \n",
       "2  ERR1073395  Female  22.41   27  United States of America   \n",
       "3  ERR1073490  Female  25.73   37  United States of America   \n",
       "4  ERR1073812  Female  29.71   55  United States of America   \n",
       "\n",
       "                                             Mesh ID Abiotrophia  \\\n",
       "0                        [D001714, D003863, D012559]         0.0   \n",
       "1  [D001327, D001714, D003863, D003920, D003967, ...         0.0   \n",
       "2  [D001327, D001714, D003863, D003920, D003967, ...         0.0   \n",
       "3                        [D001714, D003863, D012559]         0.0   \n",
       "4  [D001289, D001714, D003015, D003863, D003920, ...         0.0   \n",
       "\n",
       "  Acanthopleuribacter Acetivibrio Acetoanaerobium  ... Williamsia Wolbachia  \\\n",
       "0            0.005583         0.0             0.0  ...        0.0       0.0   \n",
       "1                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "2                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "3                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "4                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "\n",
       "  Xanthomonas Xenococcus Xenorhabdus Yersinia Yokenella Youngiibacter  \\\n",
       "0         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "1         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "2         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "3         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "4         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "\n",
       "  Zoogloea Zymomonas  \n",
       "0      0.0       0.0  \n",
       "1      0.0       0.0  \n",
       "2  0.00247       0.0  \n",
       "3      0.0       0.0  \n",
       "4      0.0       0.0  \n",
       "\n",
       "[5 rows x 930 columns]"
      ]
     },
     "execution_count": 377,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file6 = pd.read_csv('Project data files/user_selected_run_list_31dJWq43Oe.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_s_female_run_ids = run_id_file6[0]\n",
    "ERR_s_female_run_ids_drop = ERR_s_female_run_ids.drop(index=0)\n",
    "\n",
    "sch_female_species = []\n",
    "sch_female_health = []\n",
    "\n",
    "for ids in ERR_s_female_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    sch_female_species.append(genusunique)\n",
    "    sch_female_health.append(health)\n",
    "    \n",
    "\n",
    "combined_femsch_health = sch_female_health[0]\n",
    "\n",
    "for df in sch_female_health[1:]:\n",
    "    combined_femsch_health = pd.concat([combined_femsch_health, df], ignore_index=True)\n",
    "    \n",
    "combined_femsch_species = sch_female_species[0] \n",
    "\n",
    "for df1 in sch_female_species[1:]:\n",
    "    combined_femsch_species = pd.merge(combined_femsch_species, df1, on='scientific_name', how='outer')\n",
    "    combined_femsch_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_femsch = ['scientific_name'] + [col for col in combined_femsch_species.columns if col != 'scientific_name']\n",
    "combined_femsch_species = combined_femsch_species[columns_order_femsch]\n",
    "\n",
    "transposed_femsch_df = combined_femsch_species.T \n",
    "\n",
    "\n",
    "dfreset_femsch = transposed_femsch_df.reset_index()\n",
    "dfreset_femsch.columns = dfreset_femsch.iloc[0]\n",
    "\n",
    "dffinal_femsch = dfreset_femsch.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_femsch.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_femsch = dffinal_femsch['Run ID'].unique()\n",
    "\n",
    "dfmerged_femsch = []\n",
    "\n",
    "for run_id in run_ids_femsch:\n",
    "    merge = pd.merge(combined_femsch_health[combined_femsch_health['Run ID'] == run_id], \n",
    "                     dffinal_femsch[dffinal_femsch['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_femsch.append(merge)\n",
    "    \n",
    "final_femsch_merge = pd.concat(dfmerged_femsch, ignore_index=True)\n",
    "\n",
    "final_femsch_merge.to_csv('schizophrenia_female.csv', index=False)\n",
    "\n",
    "final_femsch_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0bf4e3d",
   "metadata": {},
   "source": [
    "### API - Bipolar Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "id": "b18b9945",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetivibrio</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>...</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Wenyingzhuangia</th>\n",
       "      <th>Wenzhouxiangella</th>\n",
       "      <th>Wolbachia</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Youngiibacter</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1072629</td>\n",
       "      <td>Male</td>\n",
       "      <td>17.67</td>\n",
       "      <td>64</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D007410, D008171, D012559, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00484</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1072937</td>\n",
       "      <td>Male</td>\n",
       "      <td>26.51</td>\n",
       "      <td>53</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D002318, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1073491</td>\n",
       "      <td>Male</td>\n",
       "      <td>28.08</td>\n",
       "      <td>49</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1075554</td>\n",
       "      <td>Male</td>\n",
       "      <td>30.74</td>\n",
       "      <td>62</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001714, D003863, D008881, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1075686</td>\n",
       "      <td>Male</td>\n",
       "      <td>45.84</td>\n",
       "      <td>56</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D008171, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 783 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID   Sex    BMI  Age                   Country  \\\n",
       "0  ERR1072629  Male  17.67   64  United States of America   \n",
       "1  ERR1072937  Male  26.51   53  United States of America   \n",
       "2  ERR1073491  Male  28.08   49  United States of America   \n",
       "3  ERR1075554  Male  30.74   62  United States of America   \n",
       "4  ERR1075686  Male  45.84   56  United States of America   \n",
       "\n",
       "                                             Mesh ID Abiotrophia  \\\n",
       "0  [D001714, D003863, D007410, D008171, D012559, ...         0.0   \n",
       "1               [D001714, D002318, D003863, D012559]         0.0   \n",
       "2                        [D001714, D003863, D012559]    0.003157   \n",
       "3      [D001289, D001714, D003863, D008881, D012559]         0.0   \n",
       "4               [D001714, D003863, D008171, D012559]         0.0   \n",
       "\n",
       "  Acanthopleuribacter Acetivibrio Acetoanaerobium  ... Weissella  \\\n",
       "0                 0.0         0.0             0.0  ...       0.0   \n",
       "1                 0.0         0.0             0.0  ...       0.0   \n",
       "2                 0.0         0.0             0.0  ...       0.0   \n",
       "3                 0.0         0.0             0.0  ...       0.0   \n",
       "4                 0.0         0.0             0.0  ...       0.0   \n",
       "\n",
       "  Wenyingzhuangia Wenzhouxiangella Wolbachia Xanthomonas Xenorhabdus Yersinia  \\\n",
       "0             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "1             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "2             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "3             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "4             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "\n",
       "  Youngiibacter Zoogloea Zymomonas  \n",
       "0           0.0  0.00484       0.0  \n",
       "1           0.0      0.0       0.0  \n",
       "2           0.0      0.0       0.0  \n",
       "3           0.0      0.0       0.0  \n",
       "4           0.0      0.0       0.0  \n",
       "\n",
       "[5 rows x 783 columns]"
      ]
     },
     "execution_count": 378,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file7 = pd.read_csv('Project data files/user_selected_run_list_M73s5vbyby.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_bi_male_run_ids = run_id_file7[0]\n",
    "ERR_bi_male_run_ids_drop = ERR_bi_male_run_ids.drop(index=0)\n",
    "\n",
    "bi_male_species = []\n",
    "bi_male_health = []\n",
    "\n",
    "for ids in ERR_bi_male_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    bi_male_species.append(genusunique)\n",
    "    bi_male_health.append(health)\n",
    "    \n",
    "\n",
    "combined_mbi_health = bi_male_health[0]\n",
    "\n",
    "for df in bi_male_health[1:]:\n",
    "    combined_mbi_health = pd.concat([combined_mbi_health, df], ignore_index=True)\n",
    "    \n",
    "combined_mbi_species = bi_male_species[0] \n",
    "\n",
    "for df1 in bi_male_species[1:]:\n",
    "    combined_mbi_species = pd.merge(combined_mbi_species, df1, on='scientific_name', how='outer')\n",
    "    combined_mbi_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_mbi = ['scientific_name'] + [col for col in combined_mbi_species.columns if col != 'scientific_name']\n",
    "combined_mbi_species = combined_mbi_species[columns_order_mbi]\n",
    "\n",
    "transposed_mbi_df = combined_mbi_species.T \n",
    "\n",
    "\n",
    "dfreset_mbi = transposed_mbi_df.reset_index()\n",
    "dfreset_mbi.columns = dfreset_mbi.iloc[0]\n",
    "\n",
    "dffinal_mbi = dfreset_mbi.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_mbi.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_mbi = dffinal_mbi['Run ID'].unique()\n",
    "\n",
    "dfmerged_mbi = []\n",
    "\n",
    "for run_id in run_ids_mbi:\n",
    "    merge = pd.merge(combined_mbi_health[combined_mbi_health['Run ID'] == run_id], \n",
    "                     dffinal_mbi[dffinal_mbi['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_mbi.append(merge)\n",
    "    \n",
    "final_mbi_merge = pd.concat(dfmerged_mbi, ignore_index=True)\n",
    "\n",
    "final_mbi_merge.to_csv('bipolar_male.csv', index=False)\n",
    "\n",
    "final_mbi_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47c53bb7",
   "metadata": {},
   "source": [
    "### API - Bipolar Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 379,
   "id": "813a00ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetivibrio</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>...</th>\n",
       "      <th>Williamsia</th>\n",
       "      <th>Wolbachia</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenococcus</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Yokenella</th>\n",
       "      <th>Youngiibacter</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1073023</td>\n",
       "      <td>Female</td>\n",
       "      <td>34.72</td>\n",
       "      <td>35</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005583</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1073394</td>\n",
       "      <td>Female</td>\n",
       "      <td>22.41</td>\n",
       "      <td>27</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D001714, D003863, D003920, D003967, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1073395</td>\n",
       "      <td>Female</td>\n",
       "      <td>22.41</td>\n",
       "      <td>27</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D001714, D003863, D003920, D003967, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00247</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1073490</td>\n",
       "      <td>Female</td>\n",
       "      <td>25.73</td>\n",
       "      <td>37</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1073812</td>\n",
       "      <td>Female</td>\n",
       "      <td>29.71</td>\n",
       "      <td>55</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001714, D003015, D003863, D003920, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 956 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID     Sex    BMI  Age                   Country  \\\n",
       "0  ERR1073023  Female  34.72   35  United States of America   \n",
       "1  ERR1073394  Female  22.41   27  United States of America   \n",
       "2  ERR1073395  Female  22.41   27  United States of America   \n",
       "3  ERR1073490  Female  25.73   37  United States of America   \n",
       "4  ERR1073812  Female  29.71   55  United States of America   \n",
       "\n",
       "                                             Mesh ID Abiotrophia  \\\n",
       "0                        [D001714, D003863, D012559]         0.0   \n",
       "1  [D001327, D001714, D003863, D003920, D003967, ...         0.0   \n",
       "2  [D001327, D001714, D003863, D003920, D003967, ...         0.0   \n",
       "3                        [D001714, D003863, D012559]         0.0   \n",
       "4  [D001289, D001714, D003015, D003863, D003920, ...         0.0   \n",
       "\n",
       "  Acanthopleuribacter Acetivibrio Acetoanaerobium  ... Williamsia Wolbachia  \\\n",
       "0            0.005583         0.0             0.0  ...        0.0       0.0   \n",
       "1                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "2                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "3                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "4                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "\n",
       "  Xanthomonas Xenococcus Xenorhabdus Yersinia Yokenella Youngiibacter  \\\n",
       "0         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "1         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "2         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "3         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "4         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "\n",
       "  Zoogloea Zymomonas  \n",
       "0      0.0       0.0  \n",
       "1      0.0       0.0  \n",
       "2  0.00247       0.0  \n",
       "3      0.0       0.0  \n",
       "4      0.0       0.0  \n",
       "\n",
       "[5 rows x 956 columns]"
      ]
     },
     "execution_count": 379,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file8 = pd.read_csv('Project data files/user_selected_run_list_A2hJFf9rzG.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_bi_female_run_ids = run_id_file8[0]\n",
    "ERR_bi_female_run_ids_drop = ERR_bi_female_run_ids.drop(index=0)\n",
    "\n",
    "bi_female_species = []\n",
    "bi_female_health = []\n",
    "\n",
    "for ids in ERR_bi_female_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    bi_female_species.append(genusunique)\n",
    "    bi_female_health.append(health)\n",
    "    \n",
    "\n",
    "combined_fembi_health = bi_female_health[0]\n",
    "\n",
    "for df in bi_female_health[1:]:\n",
    "    combined_fembi_health = pd.concat([combined_fembi_health, df], ignore_index=True)\n",
    "    \n",
    "combined_fembi_species = bi_female_species[0] \n",
    "\n",
    "for df1 in bi_female_species[1:]:\n",
    "    combined_fembi_species = pd.merge(combined_fembi_species, df1, on='scientific_name', how='outer')\n",
    "    combined_fembi_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_fembi = ['scientific_name'] + [col for col in combined_fembi_species.columns if col != 'scientific_name']\n",
    "combined_fembi_species = combined_fembi_species[columns_order_fembi]\n",
    "\n",
    "transposed_fembi_df = combined_fembi_species.T \n",
    "\n",
    "\n",
    "dfreset_fembi = transposed_fembi_df.reset_index()\n",
    "dfreset_fembi.columns = dfreset_fembi.iloc[0]\n",
    "\n",
    "dffinal_fembi = dfreset_fembi.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_fembi.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_fembi = dffinal_fembi['Run ID'].unique()\n",
    "\n",
    "dfmerged_fembi = []\n",
    "\n",
    "for run_id in run_ids_fembi:\n",
    "    merge = pd.merge(combined_fembi_health[combined_fembi_health['Run ID'] == run_id], \n",
    "                     dffinal_fembi[dffinal_fembi['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_fembi.append(merge)\n",
    "    \n",
    "final_fembi_merge = pd.concat(dfmerged_fembi, ignore_index=True)\n",
    "\n",
    "final_fembi_merge.to_csv('bipolar_female.csv', index=False)\n",
    "\n",
    "final_fembi_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dbe04bc8",
   "metadata": {},
   "source": [
    "### API - Epilepsy Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "id": "c90b00f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>Acholeplasma</th>\n",
       "      <th>...</th>\n",
       "      <th>Vibrio</th>\n",
       "      <th>Victivallis</th>\n",
       "      <th>Virgibacillus</th>\n",
       "      <th>Vulgatibacter</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1072832</td>\n",
       "      <td>Male</td>\n",
       "      <td>24.93</td>\n",
       "      <td>58</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D004827]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00357</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00357</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1076844</td>\n",
       "      <td>Male</td>\n",
       "      <td>21.02</td>\n",
       "      <td>36</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D003248, D004827, D008171, D013959]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1089831</td>\n",
       "      <td>Male</td>\n",
       "      <td>22.91</td>\n",
       "      <td>43</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D004827]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005888</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1089867</td>\n",
       "      <td>Male</td>\n",
       "      <td>15.62</td>\n",
       "      <td>6</td>\n",
       "      <td>Canada</td>\n",
       "      <td>[D004827]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005153</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.010306</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1089868</td>\n",
       "      <td>Male</td>\n",
       "      <td>15.62</td>\n",
       "      <td>6</td>\n",
       "      <td>Canada</td>\n",
       "      <td>[D004827]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 465 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID   Sex    BMI  Age                   Country  \\\n",
       "0  ERR1072832  Male  24.93   58  United States of America   \n",
       "1  ERR1076844  Male  21.02   36  United States of America   \n",
       "2  ERR1089831  Male  22.91   43  United States of America   \n",
       "3  ERR1089867  Male  15.62    6                    Canada   \n",
       "4  ERR1089868  Male  15.62    6                    Canada   \n",
       "\n",
       "                                Mesh ID Abiotrophia Acanthopleuribacter  \\\n",
       "0                             [D004827]         0.0                 0.0   \n",
       "1  [D003248, D004827, D008171, D013959]         0.0                 0.0   \n",
       "2                             [D004827]         0.0                 0.0   \n",
       "3                             [D004827]         0.0                 0.0   \n",
       "4                             [D004827]         0.0                 0.0   \n",
       "\n",
       "  Acetoanaerobium Acholeplasma  ... Vibrio Victivallis Virgibacillus  \\\n",
       "0         0.00357          0.0  ...    0.0     0.00357           0.0   \n",
       "1             0.0          0.0  ...    0.0         0.0           0.0   \n",
       "2             0.0          0.0  ...    0.0    0.005888           0.0   \n",
       "3             0.0          0.0  ...    0.0    0.005153           0.0   \n",
       "4             0.0          0.0  ...    0.0         0.0           0.0   \n",
       "\n",
       "  Vulgatibacter Weissella Xanthomonas Xenorhabdus Yersinia Zoogloea Zymomonas  \n",
       "0           0.0       0.0         0.0         0.0      0.0      0.0       0.0  \n",
       "1           0.0       0.0         0.0         0.0      0.0      0.0       0.0  \n",
       "2           0.0       0.0         0.0         0.0      0.0      0.0       0.0  \n",
       "3           0.0  0.010306         0.0         0.0      0.0      0.0       0.0  \n",
       "4           0.0       0.0         0.0         0.0      0.0      0.0       0.0  \n",
       "\n",
       "[5 rows x 465 columns]"
      ]
     },
     "execution_count": 380,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file9 = pd.read_csv('Project data files/user_selected_run_list_CYqElYhtzf.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_ep_male_run_ids = run_id_file9[0]\n",
    "ERR_ep_male_run_ids_drop = ERR_ep_male_run_ids.drop(index=0)\n",
    "\n",
    "ep_male_species = []\n",
    "ep_male_health = []\n",
    "\n",
    "for ids in ERR_ep_male_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    ep_male_species.append(genusunique)\n",
    "    ep_male_health.append(health)\n",
    "    \n",
    "\n",
    "combined_mep_health = ep_male_health[0]\n",
    "\n",
    "for df in ep_male_health[1:]:\n",
    "    combined_mep_health = pd.concat([combined_mep_health, df], ignore_index=True)\n",
    "    \n",
    "combined_mep_species = ep_male_species[0] \n",
    "\n",
    "for df1 in ep_male_species[1:]:\n",
    "    combined_mep_species = pd.merge(combined_mep_species, df1, on='scientific_name', how='outer')\n",
    "    combined_mep_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_mep = ['scientific_name'] + [col for col in combined_mep_species.columns if col != 'scientific_name']\n",
    "combined_mep_species = combined_mep_species[columns_order_mep]\n",
    "\n",
    "transposed_mep_df = combined_mep_species.T \n",
    "\n",
    "\n",
    "dfreset_mep = transposed_mep_df.reset_index()\n",
    "dfreset_mep.columns = dfreset_mep.iloc[0]\n",
    "\n",
    "dffinal_mep = dfreset_mep.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_mep.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_mep = dffinal_mep['Run ID'].unique()\n",
    "\n",
    "dfmerged_mep = []\n",
    "\n",
    "for run_id in run_ids_mep:\n",
    "    merge = pd.merge(combined_mep_health[combined_mep_health['Run ID'] == run_id], \n",
    "                     dffinal_mep[dffinal_mep['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_mep.append(merge)\n",
    "    \n",
    "final_mep_merge = pd.concat(dfmerged_mep, ignore_index=True)\n",
    "\n",
    "final_mep_merge.to_csv('epilepsy_male.csv', index=False)\n",
    "\n",
    "final_mep_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "731f4592",
   "metadata": {},
   "source": [
    "### API - Epilepsy Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "id": "926180fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetivibrio</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>Acetobacterium</th>\n",
       "      <th>...</th>\n",
       "      <th>Wandonia</th>\n",
       "      <th>Weeksella</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Wenzhouxiangella</th>\n",
       "      <th>Wolbachia</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Youngiibacter</th>\n",
       "      <th>Zoogloea</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1073812</td>\n",
       "      <td>Female</td>\n",
       "      <td>29.71</td>\n",
       "      <td>55</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001714, D003015, D003863, D003920, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1090471</td>\n",
       "      <td>Female</td>\n",
       "      <td>38.01</td>\n",
       "      <td>47</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D001714, D003248, D003863, D004827, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004069</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1091746</td>\n",
       "      <td>Female</td>\n",
       "      <td>31.09</td>\n",
       "      <td>59</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D003920, D004827, D013959]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001851</td>\n",
       "      <td>0.005553</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1091758</td>\n",
       "      <td>Female</td>\n",
       "      <td>21.79</td>\n",
       "      <td>62</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001327, D001714, D002318, D003248, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002317</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1160392</td>\n",
       "      <td>Female</td>\n",
       "      <td>35.19</td>\n",
       "      <td>50</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003248, D003863, D004827, D008171, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001275</td>\n",
       "      <td>0.001275</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00255</td>\n",
       "      <td>0.001275</td>\n",
       "      <td>0.001275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001275</td>\n",
       "      <td>0.005101</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 452 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID     Sex    BMI  Age                   Country  \\\n",
       "0  ERR1073812  Female  29.71   55  United States of America   \n",
       "1  ERR1090471  Female  38.01   47  United States of America   \n",
       "2  ERR1091746  Female  31.09   59  United States of America   \n",
       "3  ERR1091758  Female  21.79   62  United States of America   \n",
       "4  ERR1160392  Female  35.19   50  United States of America   \n",
       "\n",
       "                                             Mesh ID Acanthopleuribacter  \\\n",
       "0  [D001289, D001714, D003015, D003863, D003920, ...                 0.0   \n",
       "1  [D001327, D001714, D003248, D003863, D004827, ...                 0.0   \n",
       "2               [D001327, D003920, D004827, D013959]                 0.0   \n",
       "3  [D001289, D001327, D001714, D002318, D003248, ...                 0.0   \n",
       "4  [D001714, D003248, D003863, D004827, D008171, ...                 0.0   \n",
       "\n",
       "  Acetivibrio Acetoanaerobium Acetobacterium  ... Wandonia Weeksella  \\\n",
       "0         0.0             0.0            0.0  ...      0.0       0.0   \n",
       "1         0.0             0.0            0.0  ...      0.0       0.0   \n",
       "2         0.0             0.0            0.0  ...      0.0       0.0   \n",
       "3         0.0             0.0            0.0  ...      0.0       0.0   \n",
       "4         0.0        0.001275       0.001275  ...      0.0       0.0   \n",
       "\n",
       "  Weissella Wenzhouxiangella Wolbachia Xanthomonas Xenorhabdus  Yersinia  \\\n",
       "0       0.0              0.0       0.0         0.0         0.0       0.0   \n",
       "1       0.0              0.0       0.0         0.0    0.004069       0.0   \n",
       "2       0.0              0.0       0.0         0.0    0.001851  0.005553   \n",
       "3       0.0              0.0       0.0         0.0         0.0  0.002317   \n",
       "4   0.00255         0.001275  0.001275         0.0    0.001275  0.005101   \n",
       "\n",
       "  Youngiibacter Zoogloea  \n",
       "0           0.0      0.0  \n",
       "1           0.0      0.0  \n",
       "2           0.0      0.0  \n",
       "3           0.0      0.0  \n",
       "4           0.0      0.0  \n",
       "\n",
       "[5 rows x 452 columns]"
      ]
     },
     "execution_count": 381,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file10 = pd.read_csv('Project data files/user_selected_run_list_WZhhuTry3w.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_ep_female_run_ids = run_id_file10[0]\n",
    "ERR_ep_female_run_ids_drop = ERR_ep_female_run_ids.drop(index=0)\n",
    "\n",
    "ep_female_species = []\n",
    "ep_female_health = []\n",
    "\n",
    "for ids in ERR_ep_female_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    ep_female_species.append(genusunique)\n",
    "    ep_female_health.append(health)\n",
    "    \n",
    "\n",
    "combined_femep_health = ep_female_health[0]\n",
    "\n",
    "for df in ep_female_health[1:]:\n",
    "    combined_femep_health = pd.concat([combined_femep_health, df], ignore_index=True)\n",
    "    \n",
    "combined_femep_species = ep_female_species[0] \n",
    "\n",
    "for df1 in ep_female_species[1:]:\n",
    "    combined_femep_species = pd.merge(combined_femep_species, df1, on='scientific_name', how='outer')\n",
    "    combined_femep_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_femep = ['scientific_name'] + [col for col in combined_femep_species.columns if col != 'scientific_name']\n",
    "combined_femep_species = combined_femep_species[columns_order_femep]\n",
    "\n",
    "transposed_femep_df = combined_femep_species.T \n",
    "\n",
    "\n",
    "dfreset_femep = transposed_femep_df.reset_index()\n",
    "dfreset_femep.columns = dfreset_femep.iloc[0]\n",
    "\n",
    "dffinal_femep = dfreset_femep.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_femep.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_femep = dffinal_femep['Run ID'].unique()\n",
    "\n",
    "dfmerged_femep = []\n",
    "\n",
    "for run_id in run_ids_femep:\n",
    "    merge = pd.merge(combined_femep_health[combined_femep_health['Run ID'] == run_id], \n",
    "                     dffinal_femep[dffinal_femep['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_femep.append(merge)\n",
    "    \n",
    "final_femep_merge = pd.concat(dfmerged_femep, ignore_index=True)\n",
    "\n",
    "final_femep_merge.to_csv('epilepsy_female.csv', index=False)\n",
    "\n",
    "final_femep_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58625d4d",
   "metadata": {},
   "source": [
    "### API - Health Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "id": "b4cbdee4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gw/6zvc4vx95yx8vrymkvc5m8zm0000gn/T/ipykernel_49040/1574304534.py:36: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  combined_mhe_health = pd.concat([combined_mhe_health, df], ignore_index=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acetivibrio</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>Acetobacter</th>\n",
       "      <th>...</th>\n",
       "      <th>unclassified Clostridiales (miscellaneous)</th>\n",
       "      <th>unclassified Clostridiales Family XIII. Incertae Sedis</th>\n",
       "      <th>unclassified Coriobacteriaceae</th>\n",
       "      <th>unclassified Erysipelotrichaceae (miscellaneous)</th>\n",
       "      <th>unclassified Lachnospiraceae</th>\n",
       "      <th>unclassified Peptostreptococcaceae</th>\n",
       "      <th>unclassified Peptostreptococcaceae (miscellaneous)</th>\n",
       "      <th>unclassified Propionibacteriaceae</th>\n",
       "      <th>unclassified Sutterellaceae</th>\n",
       "      <th>unclassified Tissierellia</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR011245</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54.000000</td>\n",
       "      <td>Denmark</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1075960</td>\n",
       "      <td>Male</td>\n",
       "      <td>21.22</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003877</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR5078646</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.156164</td>\n",
       "      <td>Ireland</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1854747</td>\n",
       "      <td>Male</td>\n",
       "      <td>28.70</td>\n",
       "      <td>62.000000</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR688529</td>\n",
       "      <td>Male</td>\n",
       "      <td>30.90</td>\n",
       "      <td>63.000000</td>\n",
       "      <td>Austria</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00926</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02919</td>\n",
       "      <td>2.3641</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0144</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 648 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID   Sex    BMI        Age                   Country    Mesh ID  \\\n",
       "0   ERR011245  Male    NaN  54.000000                   Denmark  [D006262]   \n",
       "1  ERR1075960  Male  21.22   0.000000  United States of America  [D006262]   \n",
       "2  SRR5078646  Male    NaN   0.156164                   Ireland  [D006262]   \n",
       "3  ERR1854747  Male  28.70  62.000000            United Kingdom  [D006262]   \n",
       "4   ERR688529  Male  30.90  63.000000                   Austria  [D006262]   \n",
       "\n",
       "  Abiotrophia Acetivibrio Acetoanaerobium Acetobacter  ...  \\\n",
       "0         0.0         0.0             0.0         0.0  ...   \n",
       "1         0.0    0.003877             0.0         0.0  ...   \n",
       "2         0.0         0.0             0.0         0.0  ...   \n",
       "3         0.0         0.0             0.0         0.0  ...   \n",
       "4         0.0         0.0             0.0         0.0  ...   \n",
       "\n",
       "  unclassified Clostridiales (miscellaneous)  \\\n",
       "0                                        0.0   \n",
       "1                                        0.0   \n",
       "2                                        0.0   \n",
       "3                                        0.0   \n",
       "4                                    0.00926   \n",
       "\n",
       "  unclassified Clostridiales Family XIII. Incertae Sedis  \\\n",
       "0                                                0.0       \n",
       "1                                                0.0       \n",
       "2                                                0.0       \n",
       "3                                                0.0       \n",
       "4                                                0.0       \n",
       "\n",
       "  unclassified Coriobacteriaceae  \\\n",
       "0                            0.0   \n",
       "1                            0.0   \n",
       "2                            0.0   \n",
       "3                            0.0   \n",
       "4                            0.0   \n",
       "\n",
       "  unclassified Erysipelotrichaceae (miscellaneous)  \\\n",
       "0                                              0.0   \n",
       "1                                              0.0   \n",
       "2                                              0.0   \n",
       "3                                              0.0   \n",
       "4                                          0.02919   \n",
       "\n",
       "  unclassified Lachnospiraceae unclassified Peptostreptococcaceae  \\\n",
       "0                          0.0                                0.0   \n",
       "1                          0.0                                0.0   \n",
       "2                          0.0                                0.0   \n",
       "3                          0.0                                0.0   \n",
       "4                       2.3641                                0.0   \n",
       "\n",
       "  unclassified Peptostreptococcaceae (miscellaneous)  \\\n",
       "0                                                0.0   \n",
       "1                                                0.0   \n",
       "2                                                0.0   \n",
       "3                                                0.0   \n",
       "4                                                0.0   \n",
       "\n",
       "  unclassified Propionibacteriaceae unclassified Sutterellaceae  \\\n",
       "0                               0.0                         0.0   \n",
       "1                               0.0                         0.0   \n",
       "2                               0.0                         0.0   \n",
       "3                               0.0                         0.0   \n",
       "4                               0.0                      0.0144   \n",
       "\n",
       "  unclassified Tissierellia  \n",
       "0                       0.0  \n",
       "1                       0.0  \n",
       "2                       0.0  \n",
       "3                       0.0  \n",
       "4                       0.0  \n",
       "\n",
       "[5 rows x 648 columns]"
      ]
     },
     "execution_count": 388,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file11 = pd.read_csv('Project data files/user_selected_run_list_DW8kIIrxag.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_he_male_run_ids = run_id_file11[0]\n",
    "ERR_he_male_run_ids_drop = ERR_he_male_run_ids.drop(index=0)\n",
    "ERR_he_male_run_ids_sampled = ERR_he_male_run_ids_drop.sample(n=300, random_state=42)\n",
    "\n",
    "he_male_species = []\n",
    "he_male_health = []\n",
    "\n",
    "for ids in ERR_he_male_run_ids_sampled: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    he_male_species.append(genusunique)\n",
    "    he_male_health.append(health)\n",
    "    \n",
    "\n",
    "combined_mhe_health = he_male_health[0]\n",
    "\n",
    "for df in he_male_health[1:]:\n",
    "    combined_mhe_health = pd.concat([combined_mhe_health, df], ignore_index=True)\n",
    "    \n",
    "combined_mhe_species = he_male_species[0] \n",
    "\n",
    "for df1 in he_male_species[1:]:\n",
    "    combined_mhe_species = pd.merge(combined_mhe_species, df1, on='scientific_name', how='outer')\n",
    "    combined_mhe_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_mhe = ['scientific_name'] + [col for col in combined_mhe_species.columns if col != 'scientific_name']\n",
    "combined_mhe_species = combined_mhe_species[columns_order_mhe]\n",
    "\n",
    "transposed_mhe_df = combined_mhe_species.T \n",
    "\n",
    "\n",
    "dfreset_mhe = transposed_mhe_df.reset_index()\n",
    "dfreset_mhe.columns = dfreset_mhe.iloc[0]\n",
    "\n",
    "dffinal_mhe = dfreset_mhe.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_mhe.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_mhe = dffinal_mhe['Run ID'].unique()\n",
    "\n",
    "dfmerged_mhe = []\n",
    "\n",
    "for run_id in run_ids_mhe:\n",
    "    merge = pd.merge(combined_mhe_health[combined_mhe_health['Run ID'] == run_id], \n",
    "                     dffinal_mhe[dffinal_mhe['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_mhe.append(merge)\n",
    "    \n",
    "final_mhe_merge = pd.concat(dfmerged_mhe, ignore_index=True)\n",
    "\n",
    "final_mhe_merge.to_csv('health_male300.csv', index=False)\n",
    "\n",
    "final_mhe_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc6757f6",
   "metadata": {},
   "source": [
    "### API - Health Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "id": "89063804",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gw/6zvc4vx95yx8vrymkvc5m8zm0000gn/T/ipykernel_49040/3165339987.py:36: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  combined_femhe_health = pd.concat([combined_femhe_health, df], ignore_index=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acaryochloris</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>...</th>\n",
       "      <th>unclassified Clostridiales Family XIII. Incertae Sedis</th>\n",
       "      <th>unclassified Dermatophilaceae</th>\n",
       "      <th>unclassified Erysipelotrichaceae</th>\n",
       "      <th>unclassified Erysipelotrichaceae (miscellaneous)</th>\n",
       "      <th>unclassified Lachnospiraceae</th>\n",
       "      <th>unclassified Peptostreptococcaceae</th>\n",
       "      <th>unclassified Peptostreptococcaceae (miscellaneous)</th>\n",
       "      <th>unclassified Propionibacteriaceae</th>\n",
       "      <th>unclassified Ruminococcaceae</th>\n",
       "      <th>unclassified Sutterellaceae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR011117</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>42.0</td>\n",
       "      <td>Denmark</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1842901</td>\n",
       "      <td>Female</td>\n",
       "      <td>21.0300</td>\n",
       "      <td>63.0</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR719035</td>\n",
       "      <td>Female</td>\n",
       "      <td>19.0311</td>\n",
       "      <td>26.0</td>\n",
       "      <td>Canada</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR5648781</td>\n",
       "      <td>Female</td>\n",
       "      <td>20.0000</td>\n",
       "      <td>21.0</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.001134</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR2032350</td>\n",
       "      <td>Female</td>\n",
       "      <td>24.1300</td>\n",
       "      <td>24.0</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D006262]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 807 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID     Sex      BMI   Age                   Country    Mesh ID  \\\n",
       "0   ERR011117  Female      NaN  42.0                   Denmark  [D006262]   \n",
       "1  ERR1842901  Female  21.0300  63.0            United Kingdom  [D006262]   \n",
       "2   ERR719035  Female  19.0311  26.0                    Canada  [D006262]   \n",
       "3  SRR5648781  Female  20.0000  21.0  United States of America  [D006262]   \n",
       "4  ERR2032350  Female  24.1300  24.0  United States of America  [D006262]   \n",
       "\n",
       "  Abiotrophia Acanthopleuribacter Acaryochloris Acetoanaerobium  ...  \\\n",
       "0         0.0                 0.0           0.0             0.0  ...   \n",
       "1         0.0                 0.0           0.0             0.0  ...   \n",
       "2         0.0                 0.0           0.0             0.0  ...   \n",
       "3    0.001134                 0.0           0.0             0.0  ...   \n",
       "4         0.0                 0.0           0.0             0.0  ...   \n",
       "\n",
       "  unclassified Clostridiales Family XIII. Incertae Sedis  \\\n",
       "0                                                0.0       \n",
       "1                                                0.0       \n",
       "2                                                0.0       \n",
       "3                                                0.0       \n",
       "4                                                0.0       \n",
       "\n",
       "  unclassified Dermatophilaceae unclassified Erysipelotrichaceae  \\\n",
       "0                           0.0                              0.0   \n",
       "1                           0.0                              0.0   \n",
       "2                           0.0                              0.0   \n",
       "3                           0.0                              0.0   \n",
       "4                           0.0                              0.0   \n",
       "\n",
       "  unclassified Erysipelotrichaceae (miscellaneous)  \\\n",
       "0                                              0.0   \n",
       "1                                              0.0   \n",
       "2                                              0.0   \n",
       "3                                              0.0   \n",
       "4                                              0.0   \n",
       "\n",
       "  unclassified Lachnospiraceae unclassified Peptostreptococcaceae  \\\n",
       "0                          0.0                                0.0   \n",
       "1                          0.0                                0.0   \n",
       "2                          0.0                                0.0   \n",
       "3                          0.0                                0.0   \n",
       "4                          0.0                                0.0   \n",
       "\n",
       "  unclassified Peptostreptococcaceae (miscellaneous)  \\\n",
       "0                                                0.0   \n",
       "1                                                0.0   \n",
       "2                                                0.0   \n",
       "3                                                0.0   \n",
       "4                                                0.0   \n",
       "\n",
       "  unclassified Propionibacteriaceae unclassified Ruminococcaceae  \\\n",
       "0                               0.0                          0.0   \n",
       "1                               0.0                          0.0   \n",
       "2                               0.0                          0.0   \n",
       "3                               0.0                          0.0   \n",
       "4                               0.0                          0.0   \n",
       "\n",
       "  unclassified Sutterellaceae  \n",
       "0                         0.0  \n",
       "1                         0.0  \n",
       "2                         0.0  \n",
       "3                         0.0  \n",
       "4                         0.0  \n",
       "\n",
       "[5 rows x 807 columns]"
      ]
     },
     "execution_count": 389,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file12 = pd.read_csv('Project data files/user_selected_run_list_tYBBaBSnGX.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_he_female_run_ids = run_id_file12[0]\n",
    "ERR_he_female_run_ids_drop = ERR_he_female_run_ids.drop(index=0)\n",
    "ERR_he_female_run_ids_sampled = ERR_he_female_run_ids_drop.sample(n=300, random_state=42)\n",
    "\n",
    "he_female_species = []\n",
    "he_female_health = []\n",
    "\n",
    "for ids in ERR_he_female_run_ids_sampled: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    he_female_species.append(genusunique)\n",
    "    he_female_health.append(health)\n",
    "    \n",
    "\n",
    "combined_femhe_health = he_female_health[0]\n",
    "\n",
    "for df in he_female_health[1:]:\n",
    "    combined_femhe_health = pd.concat([combined_femhe_health, df], ignore_index=True)\n",
    "    \n",
    "combined_femhe_species = he_female_species[0] \n",
    "\n",
    "for df1 in he_female_species[1:]:\n",
    "    combined_femhe_species = pd.merge(combined_femhe_species, df1, on='scientific_name', how='outer')\n",
    "    combined_femhe_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_femhe = ['scientific_name'] + [col for col in combined_femhe_species.columns if col != 'scientific_name']\n",
    "combined_femhe_species = combined_femhe_species[columns_order_femhe]\n",
    "\n",
    "transposed_femhe_df = combined_femhe_species.T \n",
    "\n",
    "\n",
    "dfreset_femhe = transposed_femhe_df.reset_index()\n",
    "dfreset_femhe.columns = dfreset_femhe.iloc[0]\n",
    "\n",
    "dffinal_femhe = dfreset_femhe.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_femhe.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_femhe = dffinal_femhe['Run ID'].unique()\n",
    "\n",
    "dfmerged_femhe = []\n",
    "\n",
    "for run_id in run_ids_femhe:\n",
    "    merge = pd.merge(combined_femhe_health[combined_femhe_health['Run ID'] == run_id], \n",
    "                     dffinal_femhe[dffinal_femhe['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_femhe.append(merge)\n",
    "    \n",
    "final_femhe_merge = pd.concat(dfmerged_femhe, ignore_index=True)\n",
    "\n",
    "final_femhe_merge.to_csv('health_female300.csv', index=False)\n",
    "\n",
    "final_femhe_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53740bf7",
   "metadata": {},
   "source": [
    "### API - Depression Male"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "id": "41dc580e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>Acetobacter</th>\n",
       "      <th>...</th>\n",
       "      <th>Weissella</th>\n",
       "      <th>Wenyingzhuangia</th>\n",
       "      <th>Wenzhouxiangella</th>\n",
       "      <th>Wolbachia</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Youngiibacter</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1072629</td>\n",
       "      <td>Male</td>\n",
       "      <td>17.67</td>\n",
       "      <td>64</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D007410, D008171, D012559, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00484</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1072937</td>\n",
       "      <td>Male</td>\n",
       "      <td>26.51</td>\n",
       "      <td>53</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D002318, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1073491</td>\n",
       "      <td>Male</td>\n",
       "      <td>28.08</td>\n",
       "      <td>49</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1075554</td>\n",
       "      <td>Male</td>\n",
       "      <td>30.74</td>\n",
       "      <td>62</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001714, D003863, D008881, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1075686</td>\n",
       "      <td>Male</td>\n",
       "      <td>45.84</td>\n",
       "      <td>56</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D008171, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 773 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID   Sex    BMI  Age                   Country  \\\n",
       "0  ERR1072629  Male  17.67   64  United States of America   \n",
       "1  ERR1072937  Male  26.51   53  United States of America   \n",
       "2  ERR1073491  Male  28.08   49  United States of America   \n",
       "3  ERR1075554  Male  30.74   62  United States of America   \n",
       "4  ERR1075686  Male  45.84   56  United States of America   \n",
       "\n",
       "                                             Mesh ID Abiotrophia  \\\n",
       "0  [D001714, D003863, D007410, D008171, D012559, ...         0.0   \n",
       "1               [D001714, D002318, D003863, D012559]         0.0   \n",
       "2                        [D001714, D003863, D012559]    0.003157   \n",
       "3      [D001289, D001714, D003863, D008881, D012559]         0.0   \n",
       "4               [D001714, D003863, D008171, D012559]         0.0   \n",
       "\n",
       "  Acanthopleuribacter Acetoanaerobium Acetobacter  ... Weissella  \\\n",
       "0                 0.0             0.0         0.0  ...       0.0   \n",
       "1                 0.0             0.0         0.0  ...       0.0   \n",
       "2                 0.0             0.0         0.0  ...       0.0   \n",
       "3                 0.0             0.0         0.0  ...       0.0   \n",
       "4                 0.0             0.0         0.0  ...       0.0   \n",
       "\n",
       "  Wenyingzhuangia Wenzhouxiangella Wolbachia Xanthomonas Xenorhabdus Yersinia  \\\n",
       "0             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "1             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "2             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "3             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "4             0.0              0.0       0.0         0.0         0.0      0.0   \n",
       "\n",
       "  Youngiibacter Zoogloea Zymomonas  \n",
       "0           0.0  0.00484       0.0  \n",
       "1           0.0      0.0       0.0  \n",
       "2           0.0      0.0       0.0  \n",
       "3           0.0      0.0       0.0  \n",
       "4           0.0      0.0       0.0  \n",
       "\n",
       "[5 rows x 773 columns]"
      ]
     },
     "execution_count": 384,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file13 = pd.read_csv('Project data files/user_selected_run_list_KTzPlgVEYB.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_de_male_run_ids = run_id_file13[0]\n",
    "ERR_de_male_run_ids_drop = ERR_de_male_run_ids.drop(index=0)\n",
    "\n",
    "de_male_species = []\n",
    "de_male_health = []\n",
    "\n",
    "for ids in ERR_de_male_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    de_male_species.append(genusunique)\n",
    "    de_male_health.append(health)\n",
    "    \n",
    "\n",
    "combined_mde_health = de_male_health[0]\n",
    "\n",
    "for df in de_male_health[1:]:\n",
    "    combined_mde_health = pd.concat([combined_mde_health, df], ignore_index=True)\n",
    "    \n",
    "combined_mde_species = de_male_species[0] \n",
    "\n",
    "for df1 in de_male_species[1:]:\n",
    "    combined_mde_species = pd.merge(combined_mde_species, df1, on='scientific_name', how='outer')\n",
    "    combined_mde_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_mde = ['scientific_name'] + [col for col in combined_mde_species.columns if col != 'scientific_name']\n",
    "combined_mde_species = combined_mde_species[columns_order_mde]\n",
    "\n",
    "transposed_mde_df = combined_mde_species.T \n",
    "\n",
    "\n",
    "dfreset_mde = transposed_mde_df.reset_index()\n",
    "dfreset_mde.columns = dfreset_mde.iloc[0]\n",
    "\n",
    "dffinal_mde = dfreset_mde.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_mde.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_mde = dffinal_mde['Run ID'].unique()\n",
    "\n",
    "dfmerged_mde = []\n",
    "\n",
    "for run_id in run_ids_mde:\n",
    "    merge = pd.merge(combined_mde_health[combined_mde_health['Run ID'] == run_id], \n",
    "                     dffinal_mde[dffinal_mde['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_mde.append(merge)\n",
    "    \n",
    "final_mde_merge = pd.concat(dfmerged_mde, ignore_index=True)\n",
    "\n",
    "final_mde_merge.to_csv('depression_male.csv', index=False)\n",
    "\n",
    "final_mde_merge.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "98f165ed",
   "metadata": {},
   "source": [
    "### API - Depression Female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "id": "f5cc5f89",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>BMI</th>\n",
       "      <th>Age</th>\n",
       "      <th>Country</th>\n",
       "      <th>Mesh ID</th>\n",
       "      <th>Abiotrophia</th>\n",
       "      <th>Acanthopleuribacter</th>\n",
       "      <th>Acetivibrio</th>\n",
       "      <th>Acetoanaerobium</th>\n",
       "      <th>...</th>\n",
       "      <th>Williamsia</th>\n",
       "      <th>Wolbachia</th>\n",
       "      <th>Xanthomonas</th>\n",
       "      <th>Xenococcus</th>\n",
       "      <th>Xenorhabdus</th>\n",
       "      <th>Yersinia</th>\n",
       "      <th>Yokenella</th>\n",
       "      <th>Youngiibacter</th>\n",
       "      <th>Zoogloea</th>\n",
       "      <th>Zymomonas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR1073023</td>\n",
       "      <td>Female</td>\n",
       "      <td>34.72</td>\n",
       "      <td>35</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005583</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR1073394</td>\n",
       "      <td>Female</td>\n",
       "      <td>22.41</td>\n",
       "      <td>27</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D001714, D003863, D003920, D003967, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR1073395</td>\n",
       "      <td>Female</td>\n",
       "      <td>22.41</td>\n",
       "      <td>27</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001327, D001714, D003863, D003920, D003967, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00247</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR1073490</td>\n",
       "      <td>Female</td>\n",
       "      <td>25.73</td>\n",
       "      <td>37</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001714, D003863, D012559]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR1073812</td>\n",
       "      <td>Female</td>\n",
       "      <td>29.71</td>\n",
       "      <td>55</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>[D001289, D001714, D003015, D003863, D003920, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 930 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Run ID     Sex    BMI  Age                   Country  \\\n",
       "0  ERR1073023  Female  34.72   35  United States of America   \n",
       "1  ERR1073394  Female  22.41   27  United States of America   \n",
       "2  ERR1073395  Female  22.41   27  United States of America   \n",
       "3  ERR1073490  Female  25.73   37  United States of America   \n",
       "4  ERR1073812  Female  29.71   55  United States of America   \n",
       "\n",
       "                                             Mesh ID Abiotrophia  \\\n",
       "0                        [D001714, D003863, D012559]         0.0   \n",
       "1  [D001327, D001714, D003863, D003920, D003967, ...         0.0   \n",
       "2  [D001327, D001714, D003863, D003920, D003967, ...         0.0   \n",
       "3                        [D001714, D003863, D012559]         0.0   \n",
       "4  [D001289, D001714, D003015, D003863, D003920, ...         0.0   \n",
       "\n",
       "  Acanthopleuribacter Acetivibrio Acetoanaerobium  ... Williamsia Wolbachia  \\\n",
       "0            0.005583         0.0             0.0  ...        0.0       0.0   \n",
       "1                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "2                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "3                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "4                 0.0         0.0             0.0  ...        0.0       0.0   \n",
       "\n",
       "  Xanthomonas Xenococcus Xenorhabdus Yersinia Yokenella Youngiibacter  \\\n",
       "0         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "1         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "2         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "3         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "4         0.0        0.0         0.0      0.0       0.0           0.0   \n",
       "\n",
       "  Zoogloea Zymomonas  \n",
       "0      0.0       0.0  \n",
       "1      0.0       0.0  \n",
       "2  0.00247       0.0  \n",
       "3      0.0       0.0  \n",
       "4      0.0       0.0  \n",
       "\n",
       "[5 rows x 930 columns]"
      ]
     },
     "execution_count": 386,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "run_id_file14 = pd.read_csv('Project data files/user_selected_run_list_XL7BBx9USk.txt', header=None, sep='\\t')\n",
    "\n",
    "ERR_de_female_run_ids = run_id_file14[0]\n",
    "ERR_de_female_run_ids_drop = ERR_de_female_run_ids.drop(index=0)\n",
    "\n",
    "de_female_species = []\n",
    "de_female_health = []\n",
    "\n",
    "for ids in ERR_de_female_run_ids_drop: \n",
    "    query = {\"run_id\":ids}  \n",
    "    url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "    data = requests.post(url, data=json.dumps(query)).json()\n",
    "    run = data.get(\"run\")\n",
    "    genus = pd.DataFrame(data.get(\"genus\"))\n",
    "    if genus.empty:\n",
    "        continue\n",
    "    genusdrop = genus.drop(['loaded_uid', 'ncbi_taxon_id', 'taxon_rank_level'], axis=1)\n",
    "    genusrename = genusdrop.rename(columns={'relative_abundance': ids})\n",
    "    genusunique = genusrename.drop_duplicates(subset='scientific_name', keep='first')\n",
    "    sex = data['run']['sex']\n",
    "    phenotypes = data['phenotypes']\n",
    "    diseases = [item['disease'] for item in phenotypes]\n",
    "    bmi = data['run']['BMI']\n",
    "    age = data['run']['host_age']\n",
    "    country = data['run']['country']\n",
    "    health = pd.DataFrame([{'Run ID': ids, 'Sex': sex, 'BMI': bmi, 'Age': age, 'Country': country, \n",
    "                            'Mesh ID': diseases}])\n",
    "    de_female_species.append(genusunique)\n",
    "    de_female_health.append(health)\n",
    "    \n",
    "\n",
    "combined_femde_health = de_female_health[0]\n",
    "\n",
    "for df in de_female_health[1:]:\n",
    "    combined_femde_health = pd.concat([combined_femde_health, df], ignore_index=True)\n",
    "    \n",
    "combined_femde_species = de_female_species[0] \n",
    "\n",
    "for df1 in de_female_species[1:]:\n",
    "    combined_femde_species = pd.merge(combined_femde_species, df1, on='scientific_name', how='outer')\n",
    "    combined_femde_species.fillna(0, inplace=True)\n",
    "\n",
    "columns_order_femde = ['scientific_name'] + [col for col in combined_femde_species.columns if col != 'scientific_name']\n",
    "combined_femde_species = combined_femde_species[columns_order_femde]\n",
    "\n",
    "transposed_femde_df = combined_femde_species.T \n",
    "\n",
    "\n",
    "dfreset_femde = transposed_femde_df.reset_index()\n",
    "dfreset_femde.columns = dfreset_femde.iloc[0]\n",
    "\n",
    "dffinal_femde = dfreset_femde.drop(0).reset_index(drop=True)\n",
    "\n",
    "\n",
    "dffinal_femde.rename(columns={'scientific_name': 'Run ID'}, inplace=True)\n",
    "\n",
    "run_ids_femde = dffinal_femde['Run ID'].unique()\n",
    "\n",
    "dfmerged_femde = []\n",
    "\n",
    "for run_id in run_ids_femde:\n",
    "    merge = pd.merge(combined_femde_health[combined_femde_health['Run ID'] == run_id], \n",
    "                     dffinal_femde[dffinal_femde['Run ID'] == run_id], on='Run ID', how='inner')\n",
    "    dfmerged_femde.append(merge)\n",
    "    \n",
    "final_femde_merge = pd.concat(dfmerged_femde, ignore_index=True)\n",
    "\n",
    "final_femde_merge.to_csv('depression_female.csv', index=False)\n",
    "\n",
    "final_femde_merge.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c18436e7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
