{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8d51de23",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import Entrez, SeqIO\n",
    "from Bio.Blast import NCBIWWW, NCBIXML\n",
    "import re\n",
    "from collections import Counter\n",
    "import os\n",
    "from collections import Counter\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "21542af9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR10786170.1</td>\n",
       "      <td>(G, G, T, T, C, A, A, A, A, T, G, C, C, C, G, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR10786170.1</td>\n",
       "      <td>(C, N, C, C, A, T, T, A, C, C, T, T, T, C, G, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR10786170.2</td>\n",
       "      <td>(A, G, T, T, T, C, T, T, C, C, A, G, T, T, T, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR10786170.2</td>\n",
       "      <td>(T, T, T, T, G, C, A, A, G, T, A, T, T, C, A, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR10786170.3</td>\n",
       "      <td>(A, G, C, C, T, G, A, A, C, A, T, C, T, C, C, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              ID                                           Sequence\n",
       "0  ERR10786170.1  (G, G, T, T, C, A, A, A, A, T, G, C, C, C, G, ...\n",
       "1  ERR10786170.1  (C, N, C, C, A, T, T, A, C, C, T, T, T, C, G, ...\n",
       "2  ERR10786170.2  (A, G, T, T, T, C, T, T, C, C, A, G, T, T, T, ...\n",
       "3  ERR10786170.2  (T, T, T, T, G, C, A, A, G, T, A, T, T, C, A, ...\n",
       "4  ERR10786170.3  (A, G, C, C, T, G, A, A, C, A, T, C, T, C, C, ..."
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ids = []\n",
    "sequences = []\n",
    "\n",
    "fastq_file = \"dep fastq files/ERR10786170.fastq\"\n",
    "\n",
    "with open(fastq_file, \"r\") as handle:\n",
    "    for i, record in enumerate(SeqIO.parse(handle, \"fastq\")):\n",
    "        ids.append(record.id)\n",
    "        sequences.append(record.seq)\n",
    "\n",
    "df1 = pd.DataFrame({'ID': ids, 'Sequence': sequences})\n",
    "\n",
    "df1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6bf9464f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bacteria Name: gi|1887004768|gb|CP040530.1| Bacteroides thetaiotaomicron strain DSM 2079 chromosome, complete genome\n",
      "Bacteria Name: gi|2323338499|gb|CP083687.1| Bacteroides thetaiotaomicron strain VPI-2808B chromosome\n",
      "Bacteria Name: gi|29342101|gb|AE015928.1| Bacteroides thetaiotaomicron VPI-5482, complete genome\n",
      "Bacteria Name: gi|2290644306|gb|CP103155.1| Bacteroides thetaiotaomicron strain BFG-576 chromosome, complete genome\n",
      "Bacteria Name: gi|2290595203|gb|CP103214.1| Bacteroides thetaiotaomicron strain BFG-498 chromosome, complete genome\n",
      "Bacteria Name: gi|2202814649|gb|CP092641.1| Bacteroides thetaiotaomicron strain VPI 5482 chromosome, complete genome\n",
      "Bacteria Name: gi|2290770942|gb|CP103238.1| Bacteroides thetaiotaomicron strain BFG-130 chromosome, complete genome\n",
      "Bacteria Name: gi|2323328724|gb|CP083685.1| Bacteroides thetaiotaomicron strain VPI-3443 chromosome\n",
      "Bacteria Name: gi|2290599450|gb|CP103118.1| Bacteroides thetaiotaomicron strain BFG-510 chromosome, complete genome\n",
      "Bacteria Name: gi|2323333535|gb|CP083686.1| Bacteroides thetaiotaomicron strain VPI-3164 chromosome\n",
      "Bacteria Name: gi|2323314175|gb|CP083682.1| Bacteroides thetaiotaomicron strain VPI-BT8702 chromosome\n",
      "Bacteria Name: gi|2089090361|gb|CP075195.1| Bacteroides thetaiotaomicron strain CL15T119C52 chromosome, complete genome\n",
      "Bacteria Name: gi|2290509982|gb|CP103075.1| Bacteroides thetaiotaomicron strain BFG-129 chromosome, complete genome\n",
      "Bacteria Name: gi|2098064349|gb|CP081898.1| Bacteroides thetaiotaomicron strain BFG-55 chromosome, complete genome\n",
      "Bacteria Name: gi|2589387328|gb|CP134821.1| Bacteroides thetaiotaomicron strain H-2209 chromosome\n",
      "Bacteria Name: gi|2323299137|gb|CP083679.1| Bacteroides thetaiotaomicron strain WAL8713 chromosome\n",
      "Bacteria Name: gi|2290804814|gb|CP103268.1| Bacteroides thetaiotaomicron strain BFG-478 chromosome, complete genome\n",
      "Bacteria Name: gi|2089083332|gb|CP075193.1| Bacteroides thetaiotaomicron strain CL15T12C11 chromosome, complete genome\n",
      "Bacteria Name: gi|2669523424|gb|CP143941.1| Bacteroides thetaiotaomicron strain DFI.6.40 chromosome, complete genome\n",
      "Bacteria Name: gi|2323308943|gb|CP083681.1| Bacteroides thetaiotaomicron strain VPI-BTDOT2 chromosome\n",
      "Bacteria Name: gi|2323319273|gb|CP083683.1| Bacteroides thetaiotaomicron strain VPI-BT7853 chromosome\n",
      "Bacteria Name: gi|2293232970|gb|CP103354.1| Bacteroides thetaiotaomicron strain BFG-357 chromosome, complete genome\n",
      "Bacteria Name: gi|2290843172|gb|CP103283.1| Bacteroides thetaiotaomicron strain BFG-119 chromosome, complete genome\n",
      "Bacteria Name: gi|1808873978|dbj|AP022660.1| Bacteroides thetaiotaomicron F9-2 DNA, nearly complete genome\n",
      "Bacteria Name: gi|2323324036|gb|CP083684.1| Bacteroides thetaiotaomicron strain VPI-5482A chromosome\n",
      "Bacteria Name: gi|2032930700|gb|CP072242.1| Bacteroides thetaiotaomicron strain CL06T03C18 chromosome, complete genome\n",
      "Bacteria Name: gi|2290586404|gb|CP103095.1| Bacteroides thetaiotaomicron strain BFG-148 chromosome, complete genome\n",
      "Bacteria Name: gi|2290522746|gb|CP103153.1| Bacteroides thetaiotaomicron strain BFG-484 chromosome, complete genome\n",
      "Bacteria Name: gi|2323304150|gb|CP083680.1| Bacteroides thetaiotaomicron strain WAL8669 chromosome\n",
      "Bacteria Name: gi|2290796913|gb|CP103247.1| Bacteroides thetaiotaomicron strain BFG-264 chromosome, complete genome\n",
      "Bacteria Name: gi|2290531272|gb|CP103072.1| Bacteroides thetaiotaomicron strain BFG-169 chromosome, complete genome\n",
      "Bacteria Name: gi|2319002689|gb|CP107194.1| MAG: Bacteroides thetaiotaomicron isolate KR001_HAM_0006 chromosome, complete genome\n",
      "Bacteria Name: gi|2589387326|gb|CP134820.1| Bacteroides thetaiotaomicron strain H-2622 chromosome\n",
      "Bacteria Name: gi|2293264964|gb|CP103377.1| Bacteroides thetaiotaomicron strain BFG-266 chromosome, complete genome\n",
      "Bacteria Name: gi|1947644047|gb|CP065872.1| Bacteroides thetaiotaomicron strain FDAARGOS_935 chromosome\n",
      "Bacteria Name: gi|2293271900|gb|CP103375.1| Bacteroides thetaiotaomicron strain BFG-116 chromosome, complete genome\n",
      "Bacteria Name: gi|2290539801|gb|CP103077.1| Bacteroides thetaiotaomicron strain BFG-109 chromosome, complete genome\n",
      "Bacteria Name: gi|2782643052|gb|CP155529.1| Bacteroides thetaiotaomicron strain 5-290 chromosome, complete genome\n",
      "Bacteria Name: gi|2032873895|gb|CP072224.1| Bacteroides thetaiotaomicron strain CL11T00C24 chromosome, complete genome\n",
      "Bacteria Name: gi|938475369|gb|CP012937.1| Bacteroides thetaiotaomicron strain 7330 chromosome, complete genome\n",
      "Bacteria Name: gi|2290445867|gb|CP103104.1| Bacteroides thetaiotaomicron strain BFG-446 chromosome, complete genome\n",
      "Bacteria Name: gi|2293236601|gb|CP103361.1| Bacteroides thetaiotaomicron strain BFG-505 chromosome, complete genome\n",
      "Bacteria Name: gi|1828977791|gb|CP050831.1| Bacteroides faecium strain CBA7301 chromosome, complete genome\n",
      "Bacteria Name: gi|2290483394|gb|CP103071.1| Bacteroides ovatus strain BFG-464 chromosome\n",
      "Bacteria Name: gi|2290562258|gb|CP103080.1| Bacteroides ovatus strain BFG-465 chromosome\n",
      "Bacteria Name: gi|1782093953|gb|CP046397.1| Bacteroides ovatus strain FDAARGOS_733 chromosome, complete genome\n",
      "Bacteria Name: gi|2722776976|gb|CP134818.1| Bacteroides ovatus strain H-2495 chromosome\n",
      "Bacteria Name: gi|2777523482|dbj|AP031410.1| Parabacteroides goldsteinii f09 DNA, complete genome\n",
      "Bacteria Name: gi|2023091551|gb|CP064353.1| Parabacteroides goldsteinii strain MTS01 chromosome, complete genome\n",
      "Bacteria Name: gi|2023091551|gb|CP064353.1| Parabacteroides goldsteinii strain MTS01 chromosome, complete genome\n",
      "Bacteria Name: gi|2784334720|gb|CP165599.1| Parabacteroides goldsteinii strain WXC08 chromosome, complete genome\n"
     ]
    }
   ],
   "source": [
    "genomic_sequence = df1['Sequence'][0]\n",
    "\n",
    "result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", genomic_sequence)\n",
    "\n",
    "blast_records = NCBIXML.parse(result_handle)\n",
    "\n",
    "for blast_record in blast_records:\n",
    "    for alignment in blast_record.alignments:\n",
    "        for hsp in alignment.hsps:\n",
    "            if hsp.expect < 0.05:\n",
    "                print(\"Bacteria Name:\", alignment.title)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f19d06ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File Name</th>\n",
       "      <th>Bacteroides</th>\n",
       "      <th>Parabacteroides</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR10786170.fastq</td>\n",
       "      <td>0.921569</td>\n",
       "      <td>0.078431</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0          File Name Bacteroides Parabacteroides\n",
       "1  ERR10786170.fastq    0.921569        0.078431"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fastq_file = \"dep fastq files/ERR10786170.fastq\"\n",
    "\n",
    "ids = []\n",
    "sequences = []\n",
    "\n",
    "with open(fastq_file, \"r\") as handle:\n",
    "    for record in SeqIO.parse(handle, \"fastq\"):\n",
    "        ids.append(record.id)\n",
    "        sequences.append(str(record.seq))\n",
    "\n",
    "df1 = pd.DataFrame({'ID': ids, 'Sequence': sequences})\n",
    "\n",
    "genomic_sequence = df1['Sequence'][0][:1000]\n",
    "result_handle = NCBIWWW.qblast(\"blastn\", \"nt\", genomic_sequence)\n",
    "blast_records = NCBIXML.parse(result_handle)\n",
    "\n",
    "genera = []\n",
    "\n",
    "for blast_record in blast_records:\n",
    "    for alignment in blast_record.alignments:\n",
    "        for hsp in alignment.hsps:\n",
    "            if hsp.expect < 0.05:\n",
    "                order_match = re.search(r\"Uncultured (\\b[A-Z][a-z]+(?:ales|aceae|ota|etes)\\b)\", alignment.title)\n",
    "                match = re.search(r\"Uncultured ([A-Z][a-z]+) sp\", alignment.title) \n",
    "                general_match = re.search(r\"\\b([A-Z][a-z]+)\\b(?!\\s*(bacterium|organism|RNA|DNA))\", alignment.title)\n",
    "                unknown_match = re.search(r\"Uncultured (bacterium|organism)\", alignment.title, re.IGNORECASE)\n",
    "                rumen_match = re.search(r\"Uncultured rumen bacterium\", alignment.title, re.IGNORECASE)\n",
    "\n",
    "                \n",
    "                if order_match:\n",
    "                    genera.append(order_match.group(1))\n",
    "                elif match:\n",
    "                    genera.append(match.group(1))\n",
    "                elif rumen_match:\n",
    "                    genera.append(\"Rumen Bacterium\")  # Custom category for clarity\n",
    "                elif general_match:\n",
    "                    genera.append(general_match.group(1))\n",
    "                elif unknown_match:\n",
    "                    genera.append(\"Uncultured Organism\")\n",
    "\n",
    "\n",
    "\n",
    "genus_counts = Counter(genera)\n",
    "\n",
    "df = pd.DataFrame(genus_counts.items(), columns=['Genus', 'Count'])\n",
    "df['Relative Abundance'] = df['Count'] / df['Count'].sum()\n",
    "df = df.drop(columns=['Count'])\n",
    "df = df.sort_values(by='Relative Abundance', ascending=False)\n",
    "\n",
    "df = df.T\n",
    "df.insert(0, 'File Name', os.path.basename(fastq_file))\n",
    "df = df.reset_index(drop=True)\n",
    "df.columns = df.iloc[0] \n",
    "df = df.drop(0)\n",
    "df = df.rename(columns={os.path.basename(fastq_file): 'File Name'})\n",
    "\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fb92d160",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['File Name', 'Enterocloster', 'Ruminococcus', 'Eubacterium',\n",
       "       'Caproicibacterium', 'Agathobacter', 'Dorea', 'Blautia',\n",
       "       'Eisenbergiella', 'Roseburia', 'Clostridiales', 'Anaerocolumna',\n",
       "       'Lachnoclostridium', 'Marvinbryantia', 'Caudoviricetes',\n",
       "       'Rossellomorea', 'Bacteriophage', 'Candidatus', 'Bacillota',\n",
       "       'Domibacillus', 'Caproicibacter', 'Caproiciproducens'],\n",
       "      dtype='object', name=0)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9c6c8619",
   "metadata": {},
   "outputs": [],
   "source": [
    "# title = \"Bacteria Name: gi|2025053910|dbj|AP023418.1| Vescimonas coprocola MM50 DNA, complete genome\"\n",
    "\n",
    "# match = re.search(r\"\\b([A-Z][a-z]+[a-zA-Z]*)\\b(?!.*bacterium)(?!.*organism)(?!.*RNA)(?!.*DNA)\", title)\n",
    "\n",
    "# print(\"Match:\", match.group(1) if match else \"No match\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bcd4af57",
   "metadata": {},
   "outputs": [],
   "source": [
    "# title = \"gi|2025053910|dbj|AP023418.1| Vescimonas coprocola MM50 DNA, complete genome\"\n",
    "\n",
    "# # Perform regex searches\n",
    "# order_match = re.search(r\"Uncultured (\\b[A-Z][a-z]+(?:ales|aceae|ota|etes)\\b)\", title)\n",
    "# match = re.search(r\"Uncultured ([A-Z][a-z]+) sp\", title)\n",
    "# rumen_match = re.search(r\"Uncultured rumen bacterium\", title, re.IGNORECASE)\n",
    "# unknown_match = re.search(r\"Uncultured (bacterium|organism)\", title, re.IGNORECASE)\n",
    "# general_match = re.search(r\"\\b([A-Z][a-z]+)\\b(?!\\s*(bacterium|organism|RNA|DNA))\", title)\n",
    "\n",
    "# # Initialize the genera list\n",
    "# genera = []\n",
    "\n",
    "# # Check matches and append accordingly\n",
    "# if order_match:\n",
    "#     print(\"Order match found:\", order_match.group(1))\n",
    "#     genera.append(order_match.group(1))\n",
    "# elif match:\n",
    "#     print(\"Species match found:\", match.group(1))\n",
    "#     genera.append(match.group(1))\n",
    "# elif rumen_match:\n",
    "#     print(\"Rumen match found\")\n",
    "#     genera.append(\"Rumen Bacterium\")\n",
    "# elif unknown_match:\n",
    "#     print(\"Unknown match found\")\n",
    "#     genera.append(\"Uncultured Organism\")\n",
    "# elif general_match:\n",
    "#     print(\"General match found:\", general_match.group(1))\n",
    "#     genera.append(general_match.group(1))\n",
    "# else:\n",
    "#     print(\"No match found for:\", title)\n",
    "\n",
    "# # Print results\n",
    "# print(\"Final genera list:\", genera)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "362a11d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# title = \"gi|2025053910|dbj|AP023418.1| Vescimonas coprocola MM50 DNA, complete genome\"\n",
    "\n",
    "# # Simplified regex to check for \"Vescimonas\"\n",
    "# match = re.search(r\"\\b([A-Z][a-z]+)\\b\", title)\n",
    "\n",
    "# if match:\n",
    "#     print(\"Match found:\", match.group(1))\n",
    "# else:\n",
    "#     print(\"No match found\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e31c1736",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
